{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2d4add08",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "study_accession\tstudy_title\texperiment_accession\texperiment_title\texperiment_desc\torganism_taxid\torganism_name\tlibrary_name\tlibrary_strategy\tlibrary_source\tlibrary_selection\tlibrary_layout\tsample_accession\tsample_title\tbiosample\tbioproject\tinstrument\tinstrument_model\tinstrument_model_desc\ttotal_spots\ttotal_size\trun_accession\trun_total_spots\trun_total_bases\n",
      "SRP458077\tTranscriptional effects of AXL signaling and phosphorylation site mutants\tSRX21599967\tRNAseq of PC9: AXL signaling variants\tRNAseq of PC9: AXL signaling variants\t9606\tHomo sapiens\tM4EA\tRNA-Seq\tTRANSCRIPTOMIC\tOligo-dT\tPAIRED\tSRS18773628\t\tSAMN37234672\tPRJNA1012098\tIllumina NovaSeq 6000\tIllumina NovaSeq 6000\tILLUMINA\t29004879\t2823016364\tSRR25879419\t29004879\t8701463700\n",
      "SRP458077\tTranscriptional effects of AXL signaling and phosphorylation site mutants\tSRX21599966\tRNAseq of PC9: AXL signaling variants\tRNAseq of PC9: AXL signaling variants\t9606\tHomo sapiens\tM4E\tRNA-Seq\tTRANSCRIPTOMIC\tOligo-dT\tPAIRED\tSRS18773626\t\tSAMN37234671\tPRJNA1012098\tIllumina NovaSeq 6000\tIllumina NovaSeq 6000\tILLUMINA\t22293609\t2211433146\tSRR25879420\t22293609\t6688082700\n",
      "SRP458077\tTranscriptional effects of AXL signaling and phosphorylation site mutants\tSRX21599965\tRNAseq of PC9: AXL signaling variants\tRNAseq of PC9: AXL signaling variants\t9606\tHomo sapiens\tKOUT\tRNA-Seq\tTRANSCRIPTOMIC\tOligo-dT\tPAIRED\tSRS18773627\t\tSAMN37234670\tPRJNA1012098\tIllumina NovaSeq 6000\tIllumina NovaSeq 6000\tILLUMINA\t28645721\t2788092278\tSRR25879421\t28645721\t8593716300\n",
      "SRP458077\tTranscriptional effects of AXL signaling and phosphorylation site mutants\tSRX21599964\tRNAseq of PC9: AXL signaling variants\tRNAseq of PC9: AXL signaling variants\t9606\tHomo sapiens\tKOE\tRNA-Seq\tTRANSCRIPTOMIC\tOligo-dT\tPAIRED\tSRS18773625\t\tSAMN37234669\tPRJNA1012098\tIllumina NovaSeq 6000\tIllumina NovaSeq 6000\tILLUMINA\t29657040\t2889718491\tSRR25879422\t29657040\t8897112000\n",
      "SRP458077\tTranscriptional effects of AXL signaling and phosphorylation site mutants\tSRX21599963\tRNAseq of PC9: AXL signaling variants\tRNAseq of PC9: AXL signaling variants\t9606\tHomo sapiens\tKIUT\tRNA-Seq\tTRANSCRIPTOMIC\tOligo-dT\tPAIRED\tSRS18773624\t\tSAMN37234668\tPRJNA1012098\tIllumina NovaSeq 6000\tIllumina NovaSeq 6000\tILLUMINA\t29337615\t2853570488\tSRR25879423\t29337615\t8801284500\n",
      "SRP458077\tTranscriptional effects of AXL signaling and phosphorylation site mutants\tSRX21599962\tRNAseq of PC9: AXL signaling variants\tRNAseq of PC9: AXL signaling variants\t9606\tHomo sapiens\tKIEA\tRNA-Seq\tTRANSCRIPTOMIC\tOligo-dT\tPAIRED\tSRS18773623\t\tSAMN37234667\tPRJNA1012098\tIllumina NovaSeq 6000\tIllumina NovaSeq 6000\tILLUMINA\t24631418\t2454031537\tSRR25879424\t24631418\t7389425400\n",
      "SRP458077\tTranscriptional effects of AXL signaling and phosphorylation site mutants\tSRX21599961\tRNAseq of PC9: AXL signaling variants\tRNAseq of PC9: AXL signaling variants\t9606\tHomo sapiens\tKIE\tRNA-Seq\tTRANSCRIPTOMIC\tOligo-dT\tPAIRED\tSRS18773622\t\tSAMN37234666\tPRJNA1012098\tIllumina NovaSeq 6000\tIllumina NovaSeq 6000\tILLUMINA\t20604131\t2047376923\tSRR25879425\t20604131\t6181239300\n",
      "SRP458077\tTranscriptional effects of AXL signaling and phosphorylation site mutants\tSRX21599960\tRNAseq of PC9: AXL signaling variants\tRNAseq of PC9: AXL signaling variants\t9606\tHomo sapiens\tWTUT\tRNA-Seq\tTRANSCRIPTOMIC\tOligo-dT\tPAIRED\tSRS18773621\t\tSAMN37234691\tPRJNA1012098\tIllumina NovaSeq 6000\tIllumina NovaSeq 6000\tILLUMINA\t22384225\t2174522785\tSRR25879426\t22384225\t6715267500\n",
      "SRP458077\tTranscriptional effects of AXL signaling and phosphorylation site mutants\tSRX21599959\tRNAseq of PC9: AXL signaling variants\tRNAseq of PC9: AXL signaling variants\t9606\tHomo sapiens\tWTEA\tRNA-Seq\tTRANSCRIPTOMIC\tOligo-dT\tPAIRED\tSRS18773620\t\tSAMN37234690\tPRJNA1012098\tIllumina NovaSeq 6000\tIllumina NovaSeq 6000\tILLUMINA\t31241332\t3076889309\tSRR25879427\t31241332\t9372399600\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/creixell/miniconda3/envs/pysradb/bin/pysradb\", line 10, in <module>\n",
      "    sys.exit(parse_args())\n",
      "  File \"/home/creixell/miniconda3/envs/pysradb/lib/python3.10/site-packages/pysradb/cli.py\", line 1173, in parse_args\n",
      "    metadata(\n",
      "  File \"/home/creixell/miniconda3/envs/pysradb/lib/python3.10/site-packages/pysradb/cli.py\", line 76, in metadata\n",
      "    _print_save_df(df, saveto)\n",
      "  File \"/home/creixell/miniconda3/envs/pysradb/lib/python3.10/site-packages/pysradb/cli.py\", line 63, in _print_save_df\n",
      "    pretty_print_df(df)\n",
      "  File \"/home/creixell/miniconda3/envs/pysradb/lib/python3.10/site-packages/pysradb/cli.py\", line 49, in pretty_print_df\n",
      "    print(\"\\t\".join(map(str, row.tolist())))\n",
      "BrokenPipeError: [Errno 32] Broken pipe\n"
     ]
    }
   ],
   "source": [
    "! pysradb metadata PRJNA1012098 | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f240a4cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "path = \"/home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "9646560c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking download URLs\n",
      "The following files will be downloaded: \n",
      "\n",
      "run_accession study_accession experiment_accession public_url                                                                                                  download_url                                                                                            out_dir                                                    filesize\n",
      "SRR25879419   SRP458077       SRX21599967          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879419/SRR25879419.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879419/SRR25879419.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.2 GB  \n",
      "SRR25879420   SRP458077       SRX21599966          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879420/SRR25879420.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879420/SRR25879420.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 1.7 GB  \n",
      "SRR25879421   SRP458077       SRX21599965          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879421/SRR25879421.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879421/SRR25879421.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.2 GB  \n",
      "SRR25879422   SRP458077       SRX21599964          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879422/SRR25879422.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879422/SRR25879422.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.3 GB  \n",
      "SRR25879423   SRP458077       SRX21599963          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879423/SRR25879423.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879423/SRR25879423.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.3 GB  \n",
      "SRR25879424   SRP458077       SRX21599962          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879424/SRR25879424.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879424/SRR25879424.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 1.9 GB  \n",
      "SRR25879425   SRP458077       SRX21599961          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879425/SRR25879425.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879425/SRR25879425.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 1.6 GB  \n",
      "SRR25879426   SRP458077       SRX21599960          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879426/SRR25879426.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879426/SRR25879426.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 1.7 GB  \n",
      "SRR25879427   SRP458077       SRX21599959          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879427/SRR25879427.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879427/SRR25879427.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.4 GB  \n",
      "SRR25879428   SRP458077       SRX21599958          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879428/SRR25879428.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879428/SRR25879428.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.0 GB  \n",
      "SRR25879429   SRP458077       SRX21599957          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879429/SRR25879429.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879429/SRR25879429.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.1 GB  \n",
      "SRR25879430   SRP458077       SRX21599956                                                   https://sra-pub-src-1.s3.amazonaws.com/SRR25879430/M15EA_1.fq.gz.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879430/SRR25879430.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.1 GB  \n",
      "SRR25879431   SRP458077       SRX21599955          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879431/SRR25879431.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879431/SRR25879431.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.2 GB  \n",
      "SRR25879432   SRP458077       SRX21599954          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879432/SRR25879432.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879432/SRR25879432.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 1.9 GB  \n",
      "SRR25879433   SRP458077       SRX21599953          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879433/SRR25879433.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879433/SRR25879433.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.2 GB  \n",
      "SRR25879434   SRP458077       SRX21599952          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879434/SRR25879434.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879434/SRR25879434.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.0 GB  \n",
      "SRR25879435   SRP458077       SRX21599951          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879435/SRR25879435.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879435/SRR25879435.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.0 GB  \n",
      "SRR25879436   SRP458077       SRX21599950                                                 https://sra-pub-run-odp.s3.amazonaws.com/sra/SRR25879436/SRR25879436 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879436/SRR25879436.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.5 GB  \n",
      "SRR25879437   SRP458077       SRX21599949          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879437/SRR25879437.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879437/SRR25879437.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 1.8 GB  \n",
      "SRR25879438   SRP458077       SRX21599948          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879438/SRR25879438.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879438/SRR25879438.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 1.7 GB  \n",
      "SRR25879439   SRP458077       SRX21599947          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879439/SRR25879439.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879439/SRR25879439.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 1.7 GB  \n",
      "SRR25879440   SRP458077       SRX21599946                                                 https://sra-pub-run-odp.s3.amazonaws.com/sra/SRR25879440/SRR25879440 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879440/SRR25879440.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.4 GB  \n",
      "SRR25879441   SRP458077       SRX21599945          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879441/SRR25879441.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879441/SRR25879441.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.1 GB  \n",
      "SRR25879442   SRP458077       SRX21599944          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879442/SRR25879442.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879442/SRR25879442.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.1 GB  \n",
      "SRR25879443   SRP458077       SRX21599943          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879443/SRR25879443.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879443/SRR25879443.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.0 GB  \n",
      "SRR25879444   SRP458077       SRX21599942          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879444/SRR25879444.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879444/SRR25879444.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 1.7 GB  \n",
      "SRR25879445   SRP458077       SRX21599941          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879445/SRR25879445.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879445/SRR25879445.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 1.5 GB  \n",
      "SRR25879446   SRP458077       SRX21599940          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879446/SRR25879446.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879446/SRR25879446.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.1 GB  \n",
      "SRR25879447   SRP458077       SRX21599939          https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-14/SRR025/25879/SRR25879447/SRR25879447.lite.1 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR258/SRR25879447/SRR25879447.sra /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations 2.3 GB  \n",
      "\n",
      "\n",
      "Total size: 58.5 GB\n",
      "\n",
      "\n",
      "100%|███████████████████████████████████████████| 29/29 [10:16<00:00, 21.27s/it]\n"
     ]
    }
   ],
   "source": [
    "! pysradb download -y -t 8 --out-dir /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations -p PRJNA1012098"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0cba2770",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/creixell/AXLomics\n"
     ]
    }
   ],
   "source": [
    "! pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "21176749",
   "metadata": {},
   "outputs": [],
   "source": [
    "! cd /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/FASTQ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e87059e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AXL-Y726F-UT.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "spots read      : 25,070,085\n",
      "reads read      : 50,140,170\n",
      "reads written   : 50,140,170\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AXL-WT-UT.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "spots read      : 29,337,615\n",
      "reads read      : 58,675,230\n",
      "reads written   : 58,675,230\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AXL-Y821-UT.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "spots read      : 26,797,529\n",
      "reads read      : 53,595,058\n",
      "reads written   : 53,595,058\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AXL-PAR-UT.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "spots read      : 22,384,225\n",
      "reads read      : 44,768,450\n",
      "reads written   : 44,768,450\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AXL-KD-UT.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "spots read      : 26,271,970\n",
      "reads read      : 52,543,940\n",
      "reads written   : 52,543,940\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AXL-Y643F-UT.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "spots read      : 27,370,704\n",
      "reads read      : 54,741,408\n",
      "reads written   : 54,741,408\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AXL-Y698F-UT.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "spots read      : 21,515,862\n",
      "reads read      : 43,031,724\n",
      "reads written   : 43,031,724\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AXL-KO-UT.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "spots read      : 28,645,721\n",
      "reads read      : 57,291,442\n",
      "reads written   : 57,291,442\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AXL-Y750F-UT.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "spots read      : 24,376,856\n",
      "reads read      : 48,753,712\n",
      "reads written   : 48,753,712\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AXL-Y634F-UT.sra\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "\n",
    "base_dir = \"/home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/SRP458077\"\n",
    "\n",
    "for root, dirs, files in os.walk(base_dir):\n",
    "    for file in files:\n",
    "        print(file)\n",
    "        file_path = os.path.join(root, file)\n",
    "        subprocess.run([\"fasterq-dump\", file_path])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d1e7e1aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess\n",
    "\n",
    "def detect_egfr_erlotinib_resistance(\n",
    "    fastq1, fastq2, reference_fasta, star_index, egfr_bed, gtf, out_prefix, threads=8\n",
    "):\n",
    "    \"\"\"\n",
    "    Aligns RNA-seq reads, extracts EGFR region, and calls EGFR variants (including exon 19 del) using STAR and GATK Mutect2.\n",
    "    \"\"\"\n",
    "    # 1. Align reads to genome using STAR\n",
    "    star_out = f\"{out_prefix}_STAR\"\n",
    "    aligned_bam = star_out + \"_Aligned.sortedByCoord.out.bam\"\n",
    "    if not os.path.exists(aligned_bam):\n",
    "        subprocess.run([\n",
    "            \"STAR\", \"--runThreadN\", str(threads),\n",
    "            \"--genomeDir\", star_index,\n",
    "            \"--readFilesIn\", fastq1, fastq2,\n",
    "            \"--readFilesCommand\", \"zcat\" if fastq1.endswith(\".gz\") else \"cat\",\n",
    "            \"--outFileNamePrefix\", star_out + \"_\",\n",
    "            \"--outSAMtype\", \"BAM\", \"SortedByCoordinate\",\n",
    "            \"--twopassMode\", \"Basic\",\n",
    "            \"--sjdbGTFfile\", gtf\n",
    "        ], check=True)\n",
    "    else:\n",
    "        print(f\"Found existing BAM: {aligned_bam}, skipping STAR alignment.\")\n",
    "\n",
    "    # 2. Index the BAM file\n",
    "    subprocess.run([\"samtools\", \"index\", aligned_bam], check=True)\n",
    "\n",
    "    # 3. Extract only EGFR region (to speed up variant calling)\n",
    "    egfr_bam = f\"{out_prefix}.EGFR.bam\"\n",
    "    subprocess.run([\n",
    "        \"samtools\", \"view\", \"-b\", \"-L\", egfr_bed, \"-o\", egfr_bam, aligned_bam\n",
    "    ], check=True)\n",
    "    subprocess.run([\"samtools\", \"index\", egfr_bam], check=True)\n",
    "\n",
    "    # 3b. ***Add read group with sample name (required for GATK)***\n",
    "    egfr_bam_rg = f\"{out_prefix}.EGFR.rg.bam\"\n",
    "    sample_name = os.path.basename(out_prefix)   # Or set a string, e.g. \"PAR-UT\"\n",
    "    subprocess.run([\n",
    "        \"samtools\", \"addreplacerg\",\n",
    "        \"-r\", f\"@RG\\\\tID:1\\\\tSM:{sample_name}\\\\tLB:lib1\\\\tPL:ILLUMINA\",\n",
    "        \"-o\", egfr_bam_rg, egfr_bam\n",
    "    ], check=True)\n",
    "    subprocess.run([\"samtools\", \"index\", egfr_bam_rg], check=True)\n",
    "\n",
    "    # 4. Call variants with GATK Mutect2 (NO '--rna' flag, just standard Mutect2 for RNA-bam)\n",
    "    egfr_vcf = f\"{out_prefix}.EGFR.mutect2.vcf\"\n",
    "    subprocess.run([\n",
    "        \"gatk\", \"Mutect2\",\n",
    "        \"-R\", reference_fasta,\n",
    "        \"-I\", egfr_bam_rg,\n",
    "        \"-L\", egfr_bed,\n",
    "        \"-O\", egfr_vcf\n",
    "    ], check=True)\n",
    "\n",
    "    # 5. Filter Mutect2 calls\n",
    "    filtered_vcf = f\"{out_prefix}.EGFR.mutect2.filtered.vcf\"\n",
    "    subprocess.run([\n",
    "        \"gatk\", \"FilterMutectCalls\",\n",
    "        \"-R\", reference_fasta,\n",
    "        \"-V\", egfr_vcf,\n",
    "        \"-O\", filtered_vcf\n",
    "    ], check=True)\n",
    "\n",
    "    print(f\"Mutect2 variant calling for EGFR complete. See filtered VCF: {filtered_vcf}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "89e80f16",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found existing BAM: /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Results/PAR-UT_STAR_Aligned.sortedByCoord.out.bam, skipping STAR alignment.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using GATK jar /home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar Mutect2 -R /home/creixell/ERAP/Allotyping/assets/GRCh38.p14.genome.fa -I /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Results/PAR-UT.EGFR.rg.bam -L /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/egfr_exons.bed -O /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Results/PAR-UT.EGFR.mutect2.vcf\n",
      "23:02:32.912 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "23:02:33.627 INFO  Mutect2 - ------------------------------------------------------------\n",
      "23:02:33.636 INFO  Mutect2 - The Genome Analysis Toolkit (GATK) v4.6.2.0\n",
      "23:02:33.637 INFO  Mutect2 - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "23:02:33.638 INFO  Mutect2 - Executing as creixell@cb2-creixell on Linux v5.4.0-148-generic amd64\n",
      "23:02:33.638 INFO  Mutect2 - Java runtime: OpenJDK 64-Bit Server VM v17.0.15-internal+0-adhoc..src\n",
      "23:02:33.639 INFO  Mutect2 - Start Date/Time: July 17, 2025 at 11:02:32 PM PDT\n",
      "23:02:33.639 INFO  Mutect2 - ------------------------------------------------------------\n",
      "23:02:33.640 INFO  Mutect2 - ------------------------------------------------------------\n",
      "23:02:33.642 INFO  Mutect2 - HTSJDK Version: 4.2.0\n",
      "23:02:33.642 INFO  Mutect2 - Picard Version: 3.4.0\n",
      "23:02:33.643 INFO  Mutect2 - Built for Spark Version: 3.5.0\n",
      "23:02:33.648 INFO  Mutect2 - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "23:02:33.649 INFO  Mutect2 - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "23:02:33.649 INFO  Mutect2 - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "23:02:33.650 INFO  Mutect2 - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "23:02:33.650 INFO  Mutect2 - Deflater: IntelDeflater\n",
      "23:02:33.650 INFO  Mutect2 - Inflater: IntelInflater\n",
      "23:02:33.651 INFO  Mutect2 - GCS max retries/reopens: 20\n",
      "23:02:33.651 INFO  Mutect2 - Requester pays: disabled\n",
      "23:02:33.652 INFO  Mutect2 - Initializing engine\n",
      "23:02:34.103 INFO  FeatureManager - Using codec BEDCodec to read file file:///home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/egfr_exons.bed\n",
      "23:02:34.151 INFO  IntervalArgumentCollection - Processing 204435 bp from intervals\n",
      "23:02:34.167 INFO  Mutect2 - Done initializing engine\n",
      "23:02:34.202 INFO  NativeLibraryLoader - Loading libgkl_utils.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_utils.so\n",
      "23:02:34.206 INFO  NativeLibraryLoader - Loading libgkl_smithwaterman.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_smithwaterman.so\n",
      "23:02:34.209 INFO  SmithWatermanAligner - Using AVX accelerated SmithWaterman implementation\n",
      "23:02:34.232 INFO  NativeLibraryLoader - Loading libgkl_pairhmm_omp.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_pairhmm_omp.so\n",
      "23:02:34.259 INFO  IntelPairHmm - Flush-to-zero (FTZ) is enabled when running PairHMM\n",
      "23:02:34.261 INFO  IntelPairHmm - Available threads: 8\n",
      "23:02:34.261 INFO  IntelPairHmm - Requested threads: 4\n",
      "23:02:34.261 INFO  PairHMM - Using the OpenMP multi-threaded AVX-accelerated native PairHMM implementation\n",
      "23:02:34.363 INFO  ProgressMeter - Starting traversal\n",
      "23:02:34.364 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Regions Processed   Regions/Minute\n",
      "23:02:37.367 INFO  Mutect2 - 732 read(s) filtered by: MappingQualityReadFilter \n",
      "55336 read(s) filtered by: MappingQualityAvailableReadFilter \n",
      "0 read(s) filtered by: MappingQualityNotZeroReadFilter \n",
      "0 read(s) filtered by: MappedReadFilter \n",
      "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
      "0 read(s) filtered by: NotDuplicateReadFilter \n",
      "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
      "0 read(s) filtered by: NonChimericOriginalAlignmentReadFilter \n",
      "0 read(s) filtered by: NonZeroReferenceLengthAlignmentReadFilter \n",
      "0 read(s) filtered by: ReadLengthReadFilter \n",
      "0 read(s) filtered by: GoodCigarReadFilter \n",
      "0 read(s) filtered by: WellformedReadFilter \n",
      "56068 total reads filtered out of 56068 reads processed\n",
      "23:02:37.404 INFO  ProgressMeter -        chr7:55210356              0.1                   684          13652.7\n",
      "23:02:37.408 INFO  ProgressMeter - Traversal complete. Processed 684 total regions in 0.1 minutes.\n",
      "23:02:37.539 INFO  VectorLoglessPairHMM - Time spent in setup for JNI call : 0.0\n",
      "23:02:37.545 INFO  PairHMM - Total compute time in PairHMM computeLogLikelihoods() : 0.0\n",
      "23:02:37.549 INFO  SmithWatermanAligner - Total compute time in native Smith-Waterman : 0.00 sec\n",
      "23:02:37.568 INFO  Mutect2 - Shutting down engine\n",
      "[July 17, 2025 at 11:02:37 PM PDT] org.broadinstitute.hellbender.tools.walkers.mutect.Mutect2 done. Elapsed time: 0.09 minutes.\n",
      "Runtime.totalMemory()=285212672\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tool returned:\n",
      "SUCCESS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using GATK jar /home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar FilterMutectCalls -R /home/creixell/ERAP/Allotyping/assets/GRCh38.p14.genome.fa -V /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Results/PAR-UT.EGFR.mutect2.vcf -O /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Results/PAR-UT.EGFR.mutect2.filtered.vcf\n",
      "23:02:46.957 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "23:02:47.207 INFO  FilterMutectCalls - ------------------------------------------------------------\n",
      "23:02:47.212 INFO  FilterMutectCalls - The Genome Analysis Toolkit (GATK) v4.6.2.0\n",
      "23:02:47.213 INFO  FilterMutectCalls - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "23:02:47.213 INFO  FilterMutectCalls - Executing as creixell@cb2-creixell on Linux v5.4.0-148-generic amd64\n",
      "23:02:47.214 INFO  FilterMutectCalls - Java runtime: OpenJDK 64-Bit Server VM v17.0.15-internal+0-adhoc..src\n",
      "23:02:47.214 INFO  FilterMutectCalls - Start Date/Time: July 17, 2025 at 11:02:46 PM PDT\n",
      "23:02:47.214 INFO  FilterMutectCalls - ------------------------------------------------------------\n",
      "23:02:47.215 INFO  FilterMutectCalls - ------------------------------------------------------------\n",
      "23:02:47.216 INFO  FilterMutectCalls - HTSJDK Version: 4.2.0\n",
      "23:02:47.216 INFO  FilterMutectCalls - Picard Version: 3.4.0\n",
      "23:02:47.217 INFO  FilterMutectCalls - Built for Spark Version: 3.5.0\n",
      "23:02:47.221 INFO  FilterMutectCalls - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "23:02:47.222 INFO  FilterMutectCalls - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "23:02:47.222 INFO  FilterMutectCalls - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "23:02:47.222 INFO  FilterMutectCalls - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "23:02:47.223 INFO  FilterMutectCalls - Deflater: IntelDeflater\n",
      "23:02:47.223 INFO  FilterMutectCalls - Inflater: IntelInflater\n",
      "23:02:47.223 INFO  FilterMutectCalls - GCS max retries/reopens: 20\n",
      "23:02:47.223 INFO  FilterMutectCalls - Requester pays: disabled\n",
      "23:02:47.224 INFO  FilterMutectCalls - Initializing engine\n",
      "23:02:48.012 INFO  FeatureManager - Using codec VCFCodec to read file file:///home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Results/PAR-UT.EGFR.mutect2.vcf\n",
      "23:02:48.765 INFO  FilterMutectCalls - Done initializing engine\n",
      "23:02:49.466 WARN  SomaticClusteringModel - No callable sites found in Mutect stats.  Running without the full somatic clustering model.  Something is seriously wrong!\n",
      "23:02:49.518 INFO  ProgressMeter - Starting traversal\n",
      "23:02:49.524 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "23:02:49.530 INFO  FilterMutectCalls - Starting pass 0 through the variants\n",
      "23:02:49.567 INFO  FilterMutectCalls - Finished pass 0 through the variants\n",
      "23:02:50.084 INFO  FilterMutectCalls - Starting pass 1 through the variants\n",
      "23:02:50.097 INFO  FilterMutectCalls - Finished pass 1 through the variants\n",
      "23:02:50.102 INFO  FilterMutectCalls - Starting pass 2 through the variants\n",
      "23:02:50.112 INFO  FilterMutectCalls - Finished pass 2 through the variants\n",
      "23:02:50.114 INFO  FilterMutectCalls - Starting pass 3 through the variants\n",
      "23:02:50.122 INFO  FilterMutectCalls - Finished pass 3 through the variants\n",
      "23:02:50.171 INFO  FilterMutectCalls - No variants filtered by: AllowAllVariantsVariantFilter\n",
      "23:02:50.179 INFO  FilterMutectCalls - 0 read(s) filtered by: AllowAllReadsReadFilter \n",
      "\n",
      "23:02:50.181 INFO  ProgressMeter -             unmapped              0.0                     0              0.0\n",
      "23:02:50.183 INFO  ProgressMeter - Traversal complete. Processed 0 total variants in 0.0 minutes.\n",
      "23:02:50.210 INFO  FilterMutectCalls - Shutting down engine\n",
      "[July 17, 2025 at 11:02:50 PM PDT] org.broadinstitute.hellbender.tools.walkers.mutect.filtering.FilterMutectCalls done. Elapsed time: 0.06 minutes.\n",
      "Runtime.totalMemory()=285212672\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mutect2 variant calling for EGFR complete. See filtered VCF: /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Results/PAR-UT.EGFR.mutect2.filtered.vcf\n"
     ]
    }
   ],
   "source": [
    "detect_egfr_erlotinib_resistance(\n",
    "    \"/home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/AXL-PAR-UT_1.fastq\", \n",
    "    \"/home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/AXL-PAR-UT_2.fastq\", \n",
    "    '/home/creixell/ERAP/Allotyping/assets/GRCh38.p14.genome.fa', \n",
    "    '/home/creixell/ERAP/Allotyping/assets/star_index', \n",
    "    \"/home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/egfr_exons.bed\", \n",
    "    '/home/creixell/ERAP/Allotyping/assets/gencode.v44.annotation.gtf', \n",
    "    \"/home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Results/PAR-UT\",\n",
    "    threads=8\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "bf248e4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "! samtools addreplacerg \\\n",
    "    -r '@RG\\tID:1\\tSM:PAR-UT\\tLB:lib1\\tPL:ILLUMINA' \\\n",
    "    -o /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.EGFR.rg.bam \\\n",
    "    /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.EGFR.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "92224f3a",
   "metadata": {},
   "outputs": [],
   "source": [
    "! samtools index /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.EGFR.rg.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b37d244c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar HaplotypeCaller -R /home/creixell/ERAP/Allotyping/assets/GRCh38.p14.genome.fa -I /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.EGFR.rg.bam -O /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.egfr.g.vcf.gz -L chr7:55000000-55220000 --dont-use-soft-clipped-bases true --native-pair-hmm-threads 4 --standard-min-confidence-threshold-for-calling 10\n",
      "11:28:35.416 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "11:28:35.608 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
      "11:28:35.612 INFO  HaplotypeCaller - The Genome Analysis Toolkit (GATK) v4.6.2.0\n",
      "11:28:35.612 INFO  HaplotypeCaller - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "11:28:35.613 INFO  HaplotypeCaller - Executing as creixell@cb2-creixell on Linux v5.4.0-148-generic amd64\n",
      "11:28:35.613 INFO  HaplotypeCaller - Java runtime: OpenJDK 64-Bit Server VM v17.0.15-internal+0-adhoc..src\n",
      "11:28:35.613 INFO  HaplotypeCaller - Start Date/Time: July 19, 2025 at 11:28:35 AM PDT\n",
      "11:28:35.614 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
      "11:28:35.614 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
      "11:28:35.615 INFO  HaplotypeCaller - HTSJDK Version: 4.2.0\n",
      "11:28:35.615 INFO  HaplotypeCaller - Picard Version: 3.4.0\n",
      "11:28:35.616 INFO  HaplotypeCaller - Built for Spark Version: 3.5.0\n",
      "11:28:35.619 INFO  HaplotypeCaller - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "11:28:35.620 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "11:28:35.620 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "11:28:35.621 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "11:28:35.621 INFO  HaplotypeCaller - Deflater: IntelDeflater\n",
      "11:28:35.622 INFO  HaplotypeCaller - Inflater: IntelInflater\n",
      "11:28:35.622 INFO  HaplotypeCaller - GCS max retries/reopens: 20\n",
      "11:28:35.622 INFO  HaplotypeCaller - Requester pays: disabled\n",
      "11:28:35.623 INFO  HaplotypeCaller - Initializing engine\n",
      "11:28:35.895 INFO  IntervalArgumentCollection - Processing 220001 bp from intervals\n",
      "11:28:35.904 INFO  HaplotypeCaller - Done initializing engine\n",
      "11:28:35.957 INFO  NativeLibraryLoader - Loading libgkl_utils.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_utils.so\n",
      "11:28:35.961 INFO  NativeLibraryLoader - Loading libgkl_smithwaterman.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_smithwaterman.so\n",
      "11:28:35.964 INFO  SmithWatermanAligner - Using AVX accelerated SmithWaterman implementation\n",
      "11:28:35.970 INFO  HaplotypeCallerEngine - Disabling physical phasing, which is supported only for reference-model confidence output\n",
      "11:28:35.990 INFO  NativeLibraryLoader - Loading libgkl_pairhmm_omp.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_pairhmm_omp.so\n",
      "11:28:36.015 INFO  IntelPairHmm - Flush-to-zero (FTZ) is enabled when running PairHMM\n",
      "11:28:36.016 INFO  IntelPairHmm - Available threads: 8\n",
      "11:28:36.017 INFO  IntelPairHmm - Requested threads: 4\n",
      "11:28:36.017 INFO  PairHMM - Using the OpenMP multi-threaded AVX-accelerated native PairHMM implementation\n",
      "11:28:36.087 INFO  ProgressMeter - Starting traversal\n",
      "11:28:36.089 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Regions Processed   Regions/Minute\n",
      "11:28:36.923 INFO  HaplotypeCaller - 732 read(s) filtered by: MappingQualityReadFilter \n",
      "55336 read(s) filtered by: MappingQualityAvailableReadFilter \n",
      "0 read(s) filtered by: MappedReadFilter \n",
      "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
      "0 read(s) filtered by: NotDuplicateReadFilter \n",
      "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
      "0 read(s) filtered by: NonZeroReferenceLengthAlignmentReadFilter \n",
      "0 read(s) filtered by: GoodCigarReadFilter \n",
      "0 read(s) filtered by: WellformedReadFilter \n",
      "56068 total reads filtered out of 56068 reads processed\n",
      "11:28:36.927 INFO  ProgressMeter -        chr7:55218700              0.0                   734          52805.8\n",
      "11:28:36.927 INFO  ProgressMeter - Traversal complete. Processed 734 total regions in 0.0 minutes.\n",
      "11:28:36.937 INFO  VectorLoglessPairHMM - Time spent in setup for JNI call : 0.0\n",
      "11:28:36.938 INFO  PairHMM - Total compute time in PairHMM computeLogLikelihoods() : 0.0\n",
      "11:28:36.938 INFO  SmithWatermanAligner - Total compute time in native Smith-Waterman : 0.00 sec\n",
      "11:28:36.940 INFO  HaplotypeCaller - Shutting down engine\n",
      "[July 19, 2025 at 11:28:36 AM PDT] org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCaller done. Elapsed time: 0.03 minutes.\n",
      "Runtime.totalMemory()=285212672\n"
     ]
    }
   ],
   "source": [
    "!gatk HaplotypeCaller \\\n",
    "  -R /home/creixell/ERAP/Allotyping/assets/GRCh38.p14.genome.fa \\\n",
    "  -I /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.EGFR.rg.bam \\\n",
    "  -O /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.egfr.g.vcf.gz \\\n",
    "  -L chr7:55000000-55220000 \\\n",
    "  --dont-use-soft-clipped-bases true \\\n",
    "  --native-pair-hmm-threads 4 \\\n",
    "  --standard-min-confidence-threshold-for-calling 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "b8c0e222",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar HaplotypeCaller -R /home/creixell/ERAP/Allotyping/assets/GRCh38.p14.genome.fa -I /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.EGFR.rg.bam -O /tmp/all_variants.g.vcf.gz --dont-use-soft-clipped-bases true --native-pair-hmm-threads 4\n",
      "11:31:37.542 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "11:31:37.734 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
      "11:31:37.738 INFO  HaplotypeCaller - The Genome Analysis Toolkit (GATK) v4.6.2.0\n",
      "11:31:37.738 INFO  HaplotypeCaller - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "11:31:37.738 INFO  HaplotypeCaller - Executing as creixell@cb2-creixell on Linux v5.4.0-148-generic amd64\n",
      "11:31:37.738 INFO  HaplotypeCaller - Java runtime: OpenJDK 64-Bit Server VM v17.0.15-internal+0-adhoc..src\n",
      "11:31:37.739 INFO  HaplotypeCaller - Start Date/Time: July 19, 2025 at 11:31:37 AM PDT\n",
      "11:31:37.739 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
      "11:31:37.739 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
      "11:31:37.740 INFO  HaplotypeCaller - HTSJDK Version: 4.2.0\n",
      "11:31:37.740 INFO  HaplotypeCaller - Picard Version: 3.4.0\n",
      "11:31:37.740 INFO  HaplotypeCaller - Built for Spark Version: 3.5.0\n",
      "11:31:37.743 INFO  HaplotypeCaller - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "11:31:37.744 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "11:31:37.744 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "11:31:37.745 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "11:31:37.745 INFO  HaplotypeCaller - Deflater: IntelDeflater\n",
      "11:31:37.745 INFO  HaplotypeCaller - Inflater: IntelInflater\n",
      "11:31:37.745 INFO  HaplotypeCaller - GCS max retries/reopens: 20\n",
      "11:31:37.746 INFO  HaplotypeCaller - Requester pays: disabled\n",
      "11:31:37.746 INFO  HaplotypeCaller - Initializing engine\n",
      "11:31:38.047 INFO  HaplotypeCaller - Done initializing engine\n",
      "11:31:38.107 INFO  NativeLibraryLoader - Loading libgkl_utils.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_utils.so\n",
      "11:31:38.110 INFO  NativeLibraryLoader - Loading libgkl_smithwaterman.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_smithwaterman.so\n",
      "11:31:38.112 INFO  SmithWatermanAligner - Using AVX accelerated SmithWaterman implementation\n",
      "11:31:38.118 INFO  HaplotypeCallerEngine - Disabling physical phasing, which is supported only for reference-model confidence output\n",
      "11:31:38.133 INFO  NativeLibraryLoader - Loading libgkl_pairhmm_omp.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_pairhmm_omp.so\n",
      "11:31:38.155 INFO  IntelPairHmm - Flush-to-zero (FTZ) is enabled when running PairHMM\n",
      "11:31:38.156 INFO  IntelPairHmm - Available threads: 8\n",
      "11:31:38.157 INFO  IntelPairHmm - Requested threads: 4\n",
      "11:31:38.157 INFO  PairHMM - Using the OpenMP multi-threaded AVX-accelerated native PairHMM implementation\n",
      "11:31:38.226 INFO  ProgressMeter - Starting traversal\n",
      "11:31:38.227 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Regions Processed   Regions/Minute\n",
      "11:31:48.233 INFO  ProgressMeter -        chr1:35987701              0.2                119960         719760.0\n",
      "11:31:58.227 INFO  ProgressMeter -        chr1:85697701              0.3                285660         856980.0\n",
      "11:32:08.228 INFO  ProgressMeter -       chr1:138977701              0.5                463260         926520.0\n",
      "11:32:18.227 INFO  ProgressMeter -       chr1:195410701              0.7                651370         977055.0\n",
      "11:32:28.227 INFO  ProgressMeter -         chr2:3718201              0.8                842250        1010700.0\n",
      "11:32:38.227 INFO  ProgressMeter -        chr2:60982201              1.0               1033130        1033130.0\n",
      "11:32:48.228 INFO  ProgressMeter -       chr2:115429201              1.2               1214620        1041102.9\n",
      "11:32:58.227 INFO  ProgressMeter -       chr2:169006201              1.3               1393210        1044907.5\n",
      "11:33:08.227 INFO  ProgressMeter -       chr2:226054201              1.5               1583370        1055580.0\n",
      "11:33:18.227 INFO  ProgressMeter -        chr3:38676601              1.7               1766090        1059654.0\n",
      "11:33:28.227 INFO  ProgressMeter -        chr3:96150601              1.8               1957670        1067820.0\n",
      "11:33:38.227 INFO  ProgressMeter -       chr3:152502601              2.0               2145510        1072755.0\n",
      "11:33:48.227 INFO  ProgressMeter -        chr4:12520801              2.2               2339890        1079949.2\n",
      "11:33:58.227 INFO  ProgressMeter -        chr4:67615801              2.3               2523540        1081517.1\n",
      "11:34:08.228 INFO  ProgressMeter -       chr4:125470801              2.5               2716390        1086556.0\n",
      "11:34:18.227 INFO  ProgressMeter -       chr4:182833801              2.7               2907600        1090350.0\n",
      "11:34:28.227 INFO  ProgressMeter -        chr5:51002101              2.8               3102210        1094897.6\n",
      "11:34:38.227 INFO  ProgressMeter -       chr5:102500101              3.0               3273870        1091290.0\n",
      "11:34:48.227 INFO  ProgressMeter -       chr5:156908101              3.2               3455230        1091125.3\n",
      "11:34:58.227 INFO  ProgressMeter -        chr6:31163701              3.3               3641210        1092363.0\n",
      "11:35:08.228 INFO  ProgressMeter -        chr6:87692701              3.5               3829640        1094182.9\n",
      "11:35:18.227 INFO  ProgressMeter -       chr6:141380701              3.7               4008600        1093254.5\n",
      "11:35:28.228 INFO  ProgressMeter -        chr7:28177501              3.8               4200610        1095811.3\n",
      "11:35:38.227 INFO  ProgressMeter -        chr7:80977501              4.0               4376610        1094152.5\n",
      "11:35:48.227 INFO  ProgressMeter -       chr7:136072501              4.2               4560260        1094462.4\n",
      "11:35:58.227 INFO  ProgressMeter -        chr8:35796301              4.3               4757160        1097806.2\n",
      "11:36:08.227 INFO  ProgressMeter -        chr8:91239301              4.5               4941970        1098215.6\n",
      "11:36:18.227 INFO  ProgressMeter -         chr9:4159501              4.7               5135500        1100464.3\n",
      "11:36:28.228 INFO  ProgressMeter -        chr9:61120501              4.8               5325370        1101800.7\n",
      "11:36:38.227 INFO  ProgressMeter -       chr9:116767501              5.0               5510860        1102172.0\n",
      "11:36:48.227 INFO  ProgressMeter -       chr10:32105701              5.2               5689970        1101284.5\n",
      "11:36:58.227 INFO  ProgressMeter -       chr10:83141701              5.3               5860090        1098766.9\n",
      "11:37:08.227 INFO  ProgressMeter -        chr11:3815101              5.5               6041660        1098483.6\n",
      "11:37:18.227 INFO  ProgressMeter -       chr11:57425101              5.7               6220360        1097710.6\n",
      "11:37:28.227 INFO  ProgressMeter -      chr11:109715101              5.8               6394660        1096227.4\n",
      "11:37:38.227 INFO  ProgressMeter -       chr12:29381401              6.0               6577170        1096195.0\n",
      "11:37:48.228 INFO  ProgressMeter -       chr12:80705401              6.2               6748250        1094310.8\n",
      "11:37:58.227 INFO  ProgressMeter -        chr13:4309801              6.3               6937850        1095450.0\n",
      "11:38:08.227 INFO  ProgressMeter -       chr13:61492801              6.5               7128460        1096686.2\n",
      "11:38:18.227 INFO  ProgressMeter -        chr14:5526301              6.7               7323120        1098468.0\n",
      "11:38:28.227 INFO  ProgressMeter -       chr14:65073301              6.8               7521610        1100723.4\n",
      "11:38:38.227 INFO  ProgressMeter -       chr15:16439401              7.0               7716310        1102330.0\n",
      "11:38:48.227 INFO  ProgressMeter -       chr15:74954401              7.2               7911360        1103910.7\n",
      "11:38:58.227 INFO  ProgressMeter -       chr16:28505101              7.3               8096500        1104068.2\n",
      "11:39:08.227 INFO  ProgressMeter -       chr16:85928101              7.5               8287910        1105054.7\n",
      "11:39:18.227 INFO  ProgressMeter -       chr17:54170701              7.7               8483180        1106501.7\n",
      "11:39:28.227 INFO  ProgressMeter -       chr18:29161201              7.8               8677340        1107745.5\n",
      "11:39:38.227 INFO  ProgressMeter -        chr19:5196901              8.0               8865370        1108171.3\n",
      "11:39:48.227 INFO  ProgressMeter -        chr20:4188001              8.2               9057400        1109069.4\n",
      "11:39:58.228 INFO  ProgressMeter -       chr20:58680001              8.3               9239040        1108684.8\n",
      "11:40:08.227 INFO  ProgressMeter -        chr22:3466801              8.5               9425510        1108883.5\n",
      "11:40:18.227 INFO  ProgressMeter -        chrX:10242301              8.7               9617490        1109710.4\n",
      "11:40:28.228 INFO  ProgressMeter -        chrX:64251301              8.8               9797520        1109153.2\n",
      "11:40:38.227 INFO  ProgressMeter -       chrX:120792301              9.0               9985990        1109554.4\n",
      "11:40:48.228 INFO  ProgressMeter -        chrY:17719201              9.2              10162550        1108641.8\n",
      "11:40:58.227 INFO  ProgressMeter -     KV880764.1:75901              9.3              10353680        1109322.9\n",
      "11:41:08.227 INFO  ProgressMeter -   KI270905.1:3966601              9.5              10550960        1110627.4\n",
      "11:41:18.227 INFO  ProgressMeter -    KQ983256.1:299701              9.7              10726740        1109662.8\n",
      "11:41:28.227 INFO  ProgressMeter -    KI270821.1:886201              9.8              10920440        1110553.2\n",
      "11:41:30.885 INFO  HaplotypeCaller - 732 read(s) filtered by: MappingQualityReadFilter \n",
      "55336 read(s) filtered by: MappingQualityAvailableReadFilter \n",
      "0 read(s) filtered by: MappedReadFilter \n",
      "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
      "0 read(s) filtered by: NotDuplicateReadFilter \n",
      "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
      "0 read(s) filtered by: NonZeroReferenceLengthAlignmentReadFilter \n",
      "0 read(s) filtered by: GoodCigarReadFilter \n",
      "0 read(s) filtered by: WellformedReadFilter \n",
      "56068 total reads filtered out of 56068 reads processed\n",
      "11:41:30.886 INFO  ProgressMeter -     KI270757.1:69001              9.9              10972287        1110821.5\n",
      "11:41:30.886 INFO  ProgressMeter - Traversal complete. Processed 10972287 total regions in 9.9 minutes.\n",
      "11:41:30.896 INFO  VectorLoglessPairHMM - Time spent in setup for JNI call : 0.0\n",
      "11:41:30.897 INFO  PairHMM - Total compute time in PairHMM computeLogLikelihoods() : 0.0\n",
      "11:41:30.897 INFO  SmithWatermanAligner - Total compute time in native Smith-Waterman : 0.00 sec\n",
      "11:41:30.900 INFO  HaplotypeCaller - Shutting down engine\n",
      "[July 19, 2025 at 11:41:30 AM PDT] org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCaller done. Elapsed time: 9.89 minutes.\n",
      "Runtime.totalMemory()=1962934272\n"
     ]
    }
   ],
   "source": [
    "! gatk HaplotypeCaller \\\n",
    "  -R /home/creixell/ERAP/Allotyping/assets/GRCh38.p14.genome.fa \\\n",
    "  -I /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.EGFR.rg.bam \\\n",
    "  -O /tmp/all_variants.g.vcf.gz \\\n",
    "  --dont-use-soft-clipped-bases true \\\n",
    "  --native-pair-hmm-threads 4\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "4bdde4af",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using GATK jar /home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar HaplotypeCaller -R /home/creixell/ERAP/Allotyping/assets/GRCh38.p14.genome.fa -I /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.EGFR.rg.bam -O /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.egfr.del19.vcf.gz -L chr7:55174739-55174819 --dont-use-soft-clipped-bases false --min-pruning 1 --standard-min-confidence-threshold-for-calling 10 --native-pair-hmm-threads 4\n",
      "11:50:18.245 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "11:50:18.456 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
      "11:50:18.461 INFO  HaplotypeCaller - The Genome Analysis Toolkit (GATK) v4.6.2.0\n",
      "11:50:18.461 INFO  HaplotypeCaller - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "11:50:18.461 INFO  HaplotypeCaller - Executing as creixell@cb2-creixell on Linux v5.4.0-148-generic amd64\n",
      "11:50:18.461 INFO  HaplotypeCaller - Java runtime: OpenJDK 64-Bit Server VM v17.0.15-internal+0-adhoc..src\n",
      "11:50:18.462 INFO  HaplotypeCaller - Start Date/Time: July 19, 2025 at 11:50:18 AM PDT\n",
      "11:50:18.462 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
      "11:50:18.462 INFO  HaplotypeCaller - ------------------------------------------------------------\n",
      "11:50:18.463 INFO  HaplotypeCaller - HTSJDK Version: 4.2.0\n",
      "11:50:18.464 INFO  HaplotypeCaller - Picard Version: 3.4.0\n",
      "11:50:18.464 INFO  HaplotypeCaller - Built for Spark Version: 3.5.0\n",
      "11:50:18.467 INFO  HaplotypeCaller - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "11:50:18.468 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "11:50:18.468 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "11:50:18.468 INFO  HaplotypeCaller - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "11:50:18.469 INFO  HaplotypeCaller - Deflater: IntelDeflater\n",
      "11:50:18.469 INFO  HaplotypeCaller - Inflater: IntelInflater\n",
      "11:50:18.469 INFO  HaplotypeCaller - GCS max retries/reopens: 20\n",
      "11:50:18.469 INFO  HaplotypeCaller - Requester pays: disabled\n",
      "11:50:18.470 INFO  HaplotypeCaller - Initializing engine\n",
      "11:50:18.772 INFO  IntervalArgumentCollection - Processing 81 bp from intervals\n",
      "11:50:18.782 INFO  HaplotypeCaller - Done initializing engine\n",
      "11:50:18.852 INFO  NativeLibraryLoader - Loading libgkl_utils.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_utils.so\n",
      "11:50:18.856 INFO  NativeLibraryLoader - Loading libgkl_smithwaterman.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_smithwaterman.so\n",
      "11:50:18.859 INFO  SmithWatermanAligner - Using AVX accelerated SmithWaterman implementation\n",
      "11:50:18.864 INFO  HaplotypeCallerEngine - Disabling physical phasing, which is supported only for reference-model confidence output\n",
      "11:50:18.882 INFO  NativeLibraryLoader - Loading libgkl_pairhmm_omp.so from jar:file:/home/creixell/miniconda3/envs/pysradb/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_pairhmm_omp.so\n",
      "11:50:18.906 INFO  IntelPairHmm - Flush-to-zero (FTZ) is enabled when running PairHMM\n",
      "11:50:18.908 INFO  IntelPairHmm - Available threads: 8\n",
      "11:50:18.908 INFO  IntelPairHmm - Requested threads: 4\n",
      "11:50:18.908 INFO  PairHMM - Using the OpenMP multi-threaded AVX-accelerated native PairHMM implementation\n",
      "11:50:18.972 INFO  ProgressMeter - Starting traversal\n",
      "11:50:18.974 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Regions Processed   Regions/Minute\n",
      "11:50:19.115 INFO  HaplotypeCaller - 38 read(s) filtered by: MappingQualityReadFilter \n",
      "1328 read(s) filtered by: MappingQualityAvailableReadFilter \n",
      "0 read(s) filtered by: MappedReadFilter \n",
      "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
      "0 read(s) filtered by: NotDuplicateReadFilter \n",
      "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
      "0 read(s) filtered by: NonZeroReferenceLengthAlignmentReadFilter \n",
      "0 read(s) filtered by: GoodCigarReadFilter \n",
      "0 read(s) filtered by: WellformedReadFilter \n",
      "1366 total reads filtered out of 1366 reads processed\n",
      "11:50:19.116 INFO  ProgressMeter -             unmapped              0.0                     1            425.5\n",
      "11:50:19.116 INFO  ProgressMeter - Traversal complete. Processed 1 total regions in 0.0 minutes.\n",
      "11:50:19.125 INFO  VectorLoglessPairHMM - Time spent in setup for JNI call : 0.0\n",
      "11:50:19.126 INFO  PairHMM - Total compute time in PairHMM computeLogLikelihoods() : 0.0\n",
      "11:50:19.126 INFO  SmithWatermanAligner - Total compute time in native Smith-Waterman : 0.00 sec\n",
      "11:50:19.128 INFO  HaplotypeCaller - Shutting down engine\n",
      "[July 19, 2025 at 11:50:19 AM PDT] org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCaller done. Elapsed time: 0.02 minutes.\n",
      "Runtime.totalMemory()=285212672\n"
     ]
    }
   ],
   "source": [
    "!gatk HaplotypeCaller \\\n",
    "  -R /home/creixell/ERAP/Allotyping/assets/GRCh38.p14.genome.fa \\\n",
    "  -I /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.EGFR.rg.bam \\\n",
    "  -O /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.egfr.del19.vcf.gz \\\n",
    "  -L chr7:55174739-55174819 \\\n",
    "  --dont-use-soft-clipped-bases false \\\n",
    "  --min-pruning 1 \\\n",
    "  --standard-min-confidence-threshold-for-calling 10 \\\n",
    "  --native-pair-hmm-threads 4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "f292f11f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##fileformat=VCFv4.2\n",
      "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
      "##FILTER=<ID=LowQual,Description=\"Low quality\">\n",
      "##FORMAT=<ID=AD,Number=R,Type=Integer,Description=\"Allelic depths for the ref and alt alleles in the order listed\">\n",
      "##FORMAT=<ID=DP,Number=1,Type=Integer,Description=\"Approximate read depth (reads with MQ=255 or with bad mates are filtered)\">\n",
      "##FORMAT=<ID=GQ,Number=1,Type=Integer,Description=\"Genotype Quality\">\n",
      "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">\n",
      "##FORMAT=<ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "##GATKCommandLine=<ID=HaplotypeCaller,CommandLine=\"HaplotypeCaller --standard-min-confidence-threshold-for-calling 10.0 --min-pruning 1 --native-pair-hmm-threads 4 --dont-use-soft-clipped-bases false --output /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.egfr.del19.vcf.gz --intervals chr7:55174739-55174819 --input /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.EGFR.rg.bam --reference /home/creixell/ERAP/Allotyping/assets/GRCh38.p14.genome.fa --use-posteriors-to-calculate-qual false --dont-use-dragstr-priors false --use-new-qual-calculator true --annotate-with-num-discovered-alleles false --heterozygosity 0.001 --indel-heterozygosity 1.25E-4 --heterozygosity-stdev 0.01 --max-alternate-alleles 6 --max-genotype-count 1024 --sample-ploidy 2 --num-reference-samples-if-no-call 0 --genotype-assignment-method USE_PLS_TO_ASSIGN --contamination-fraction-to-filter 0.0 --output-mode EMIT_VARIANTS_ONLY --all-site-pls false --flow-likelihood-parallel-threads 0 --flow-likelihood-optimized-comp false --trim-to-haplotype true --exact-matching false --flow-use-t0-tag false --flow-remove-non-single-base-pair-indels false --flow-remove-one-zero-probs false --flow-quantization-bins 121 --flow-fill-empty-bins-value 0.001 --flow-symmetric-indel-probs false --flow-report-insertion-or-deletion false --flow-disallow-probs-larger-than-call false --flow-lump-probs false --flow-retain-max-n-probs-base-format false --flow-probability-scaling-factor 10 --flow-order-cycle-length 4 --keep-boundary-flows false --gvcf-gq-bands 1 --gvcf-gq-bands 2 --gvcf-gq-bands 3 --gvcf-gq-bands 4 --gvcf-gq-bands 5 --gvcf-gq-bands 6 --gvcf-gq-bands 7 --gvcf-gq-bands 8 --gvcf-gq-bands 9 --gvcf-gq-bands 10 --gvcf-gq-bands 11 --gvcf-gq-bands 12 --gvcf-gq-bands 13 --gvcf-gq-bands 14 --gvcf-gq-bands 15 --gvcf-gq-bands 16 --gvcf-gq-bands 17 --gvcf-gq-bands 18 --gvcf-gq-bands 19 --gvcf-gq-bands 20 --gvcf-gq-bands 21 --gvcf-gq-bands 22 --gvcf-gq-bands 23 --gvcf-gq-bands 24 --gvcf-gq-bands 25 --gvcf-gq-bands 26 --gvcf-gq-bands 27 --gvcf-gq-bands 28 --gvcf-gq-bands 29 --gvcf-gq-bands 30 --gvcf-gq-bands 31 --gvcf-gq-bands 32 --gvcf-gq-bands 33 --gvcf-gq-bands 34 --gvcf-gq-bands 35 --gvcf-gq-bands 36 --gvcf-gq-bands 37 --gvcf-gq-bands 38 --gvcf-gq-bands 39 --gvcf-gq-bands 40 --gvcf-gq-bands 41 --gvcf-gq-bands 42 --gvcf-gq-bands 43 --gvcf-gq-bands 44 --gvcf-gq-bands 45 --gvcf-gq-bands 46 --gvcf-gq-bands 47 --gvcf-gq-bands 48 --gvcf-gq-bands 49 --gvcf-gq-bands 50 --gvcf-gq-bands 51 --gvcf-gq-bands 52 --gvcf-gq-bands 53 --gvcf-gq-bands 54 --gvcf-gq-bands 55 --gvcf-gq-bands 56 --gvcf-gq-bands 57 --gvcf-gq-bands 58 --gvcf-gq-bands 59 --gvcf-gq-bands 60 --gvcf-gq-bands 70 --gvcf-gq-bands 80 --gvcf-gq-bands 90 --gvcf-gq-bands 99 --floor-blocks false --indel-size-to-eliminate-in-ref-model 10 --disable-optimizations false --dragen-mode false --dragen-378-concordance-mode false --flow-mode NONE --apply-bqd false --apply-frd false --disable-spanning-event-genotyping false --transform-dragen-mapping-quality false --mapping-quality-threshold-for-genotyping 20 --max-effective-depth-adjustment-for-frd 0 --just-determine-active-regions false --dont-genotype false --do-not-run-physical-phasing false --do-not-correct-overlapping-quality false --use-filtered-reads-for-annotations false --use-flow-aligner-for-stepwise-hc-filtering false --adaptive-pruning false --do-not-recover-dangling-branches false --recover-dangling-heads false --kmer-size 10 --kmer-size 25 --dont-increase-kmer-sizes-for-cycles false --allow-non-unique-kmers-in-ref false --num-pruning-samples 1 --min-dangling-branch-length 4 --recover-all-dangling-branches false --max-num-haplotypes-in-population 128 --adaptive-pruning-initial-error-rate 0.001 --pruning-lod-threshold 2.302585092994046 --pruning-seeding-lod-threshold 9.210340371976184 --max-unpruned-variants 100 --linked-de-bruijn-graph false --disable-artificial-haplotype-recovery false --enable-legacy-graph-cycle-detection false --debug-assembly false --debug-graph-transformations false --capture-assembly-failure-bam false --num-matching-bases-in-dangling-end-to-recover -1 --error-correction-log-odds -Infinity --error-correct-reads false --kmer-length-for-read-error-correction 25 --min-observations-for-kmer-to-be-solid 20 --likelihood-calculation-engine PairHMM --base-quality-score-threshold 18 --dragstr-het-hom-ratio 2 --dont-use-dragstr-pair-hmm-scores false --pair-hmm-gap-continuation-penalty 10 --expected-mismatch-rate-for-read-disqualification 0.02 --pair-hmm-implementation FASTEST_AVAILABLE --pcr-indel-model CONSERVATIVE --phred-scaled-global-read-mismapping-rate 45 --disable-symmetric-hmm-normalizing false --disable-cap-base-qualities-to-map-quality false --enable-dynamic-read-disqualification-for-genotyping false --dynamic-read-disqualification-threshold 1.0 --native-pair-hmm-use-double-precision false --flow-hmm-engine-min-indel-adjust 6 --flow-hmm-engine-flat-insertion-penatly 45 --flow-hmm-engine-flat-deletion-penatly 45 --pileup-detection false --use-pdhmm false --use-pdhmm-overlap-optimization false --make-determined-haps-from-pd-code false --print-pileupcalling-status false --fallback-gga-if-pdhmm-fails true --pileup-detection-enable-indel-pileup-calling false --pileup-detection-active-region-phred-threshold 0.0 --num-artificial-haplotypes-to-add-per-allele 5 --artifical-haplotype-filtering-kmer-size 10 --pileup-detection-snp-alt-threshold 0.1 --pileup-detection-indel-alt-threshold 0.1 --pileup-detection-absolute-alt-depth 0.0 --pileup-detection-snp-adjacent-to-assembled-indel-range 5 --pileup-detection-snp-basequality-filter 12 --pileup-detection-bad-read-tolerance 0.0 --pileup-detection-proper-pair-read-badness true --pileup-detection-edit-distance-read-badness-threshold 0.08 --pileup-detection-chimeric-read-badness true --pileup-detection-template-mean-badness-threshold 0.0 --pileup-detection-template-std-badness-threshold 0.0 --pileup-detection-filter-assembly-alt-bad-read-tolerance 0.0 --pileup-detection-edit-distance-read-badness-for-assembly-filtering-threshold 0.12 --bam-writer-type CALLED_HAPLOTYPES --override-fragment-softclip-check false --min-base-quality-score 10 --smith-waterman FASTEST_AVAILABLE --emit-ref-confidence NONE --max-mnp-distance 0 --force-call-filtered-alleles false --reference-model-deletion-quality 30 --soft-clip-low-quality-ends false --allele-informative-reads-overlap-margin 2 --smith-waterman-dangling-end-match-value 25 --smith-waterman-dangling-end-mismatch-penalty -50 --smith-waterman-dangling-end-gap-open-penalty -110 --smith-waterman-dangling-end-gap-extend-penalty -6 --smith-waterman-haplotype-to-reference-match-value 200 --smith-waterman-haplotype-to-reference-mismatch-penalty -150 --smith-waterman-haplotype-to-reference-gap-open-penalty -260 --smith-waterman-haplotype-to-reference-gap-extend-penalty -11 --smith-waterman-read-to-haplotype-match-value 10 --smith-waterman-read-to-haplotype-mismatch-penalty -15 --smith-waterman-read-to-haplotype-gap-open-penalty -30 --smith-waterman-read-to-haplotype-gap-extend-penalty -5 --flow-assembly-collapse-hmer-size 0 --flow-assembly-collapse-partial-mode false --flow-filter-alleles false --flow-filter-alleles-qual-threshold 30.0 --flow-filter-alleles-sor-threshold 3.0 --flow-filter-lone-alleles false --flow-filter-alleles-debug-graphs false --min-assembly-region-size 50 --max-assembly-region-size 300 --active-probability-threshold 0.002 --max-prob-propagation-distance 50 --force-active false --assembly-region-padding 100 --padding-around-indels 75 --padding-around-snps 20 --padding-around-strs 75 --max-extension-into-assembly-region-padding-legacy 25 --max-reads-per-alignment-start 50 --enable-legacy-assembly-region-trimming false --interval-set-rule UNION --interval-padding 0 --interval-exclusion-padding 0 --interval-merging-rule ALL --read-validation-stringency SILENT --seconds-between-progress-updates 10.0 --disable-sequence-dictionary-validation false --create-output-bam-index true --create-output-bam-md5 false --create-output-variant-index true --create-output-variant-md5 false --max-variants-per-shard 0 --lenient false --add-output-sam-program-record true --add-output-vcf-command-line true --cloud-prefetch-buffer 40 --cloud-index-prefetch-buffer -1 --disable-bam-index-caching false --sites-only-vcf-output false --help false --version false --showHidden false --verbosity INFO --QUIET false --use-jdk-deflater false --use-jdk-inflater false --gcs-max-retries 20 --gcs-project-for-requester-pays  --disable-tool-default-read-filters false --minimum-mapping-quality 20 --disable-tool-default-annotations false --enable-all-annotations false --allow-old-rms-mapping-quality-annotation-data false\",Version=\"4.6.2.0\",Date=\"July 19, 2025 at 11:50:18 AM PDT\">\n",
      "##INFO=<ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "##INFO=<ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "##INFO=<ID=AN,Number=1,Type=Integer,Description=\"Total number of alleles in called genotypes\">\n",
      "##INFO=<ID=BaseQRankSum,Number=1,Type=Float,Description=\"Z-score from Wilcoxon rank sum test of Alt Vs. Ref base qualities\">\n",
      "##INFO=<ID=DP,Number=1,Type=Integer,Description=\"Approximate read depth; some reads may have been filtered\">\n",
      "##INFO=<ID=ExcessHet,Number=1,Type=Float,Description=\"Phred-scaled p-value for exact test of excess heterozygosity\">\n",
      "##INFO=<ID=FS,Number=1,Type=Float,Description=\"Phred-scaled p-value using Fisher's exact test to detect strand bias\">\n",
      "##INFO=<ID=InbreedingCoeff,Number=1,Type=Float,Description=\"Inbreeding coefficient as estimated from the genotype likelihoods per-sample when compared against the Hardy-Weinberg expectation\">\n",
      "##INFO=<ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "##INFO=<ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "##INFO=<ID=MQ,Number=1,Type=Float,Description=\"RMS Mapping Quality\">\n",
      "##INFO=<ID=MQRankSum,Number=1,Type=Float,Description=\"Z-score From Wilcoxon rank sum test of Alt vs. Ref read mapping qualities\">\n",
      "##INFO=<ID=QD,Number=1,Type=Float,Description=\"Variant Confidence/Quality by Depth\">\n",
      "##INFO=<ID=ReadPosRankSum,Number=1,Type=Float,Description=\"Z-score from Wilcoxon rank sum test of Alt vs. Ref read position bias\">\n",
      "##INFO=<ID=SOR,Number=1,Type=Float,Description=\"Symmetric Odds Ratio of 2x2 contingency table to detect strand bias\">\n",
      "##contig=<ID=chr1,length=248956422>\n",
      "##contig=<ID=chr2,length=242193529>\n",
      "##contig=<ID=chr3,length=198295559>\n",
      "##contig=<ID=chr4,length=190214555>\n",
      "##contig=<ID=chr5,length=181538259>\n",
      "##contig=<ID=chr6,length=170805979>\n",
      "##contig=<ID=chr7,length=159345973>\n",
      "##contig=<ID=chr8,length=145138636>\n",
      "##contig=<ID=chr9,length=138394717>\n",
      "##contig=<ID=chr10,length=133797422>\n",
      "##contig=<ID=chr11,length=135086622>\n",
      "##contig=<ID=chr12,length=133275309>\n",
      "##contig=<ID=chr13,length=114364328>\n",
      "##contig=<ID=chr14,length=107043718>\n",
      "##contig=<ID=chr15,length=101991189>\n",
      "##contig=<ID=chr16,length=90338345>\n",
      "##contig=<ID=chr17,length=83257441>\n",
      "##contig=<ID=chr18,length=80373285>\n",
      "##contig=<ID=chr19,length=58617616>\n",
      "##contig=<ID=chr20,length=64444167>\n",
      "##contig=<ID=chr21,length=46709983>\n",
      "##contig=<ID=chr22,length=50818468>\n",
      "##contig=<ID=chrX,length=156040895>\n",
      "##contig=<ID=chrY,length=57227415>\n",
      "##contig=<ID=chrM,length=16569>\n",
      "##contig=<ID=MU273365.1,length=482250>\n",
      "##contig=<ID=MU273352.1,length=34400>\n",
      "##contig=<ID=MU273360.1,length=39290>\n",
      "##contig=<ID=KQ759759.2,length=204999>\n",
      "##contig=<ID=ML143376.1,length=493165>\n",
      "##contig=<ID=MU273366.1,length=569668>\n",
      "##contig=<ID=KN538364.1,length=415308>\n",
      "##contig=<ID=ML143355.1,length=292944>\n",
      "##contig=<ID=ML143348.1,length=125549>\n",
      "##contig=<ID=ML143347.1,length=176674>\n",
      "##contig=<ID=ML143346.1,length=53476>\n",
      "##contig=<ID=ML143352.1,length=254759>\n",
      "##contig=<ID=KQ759762.2,length=101040>\n",
      "##contig=<ID=ML143375.1,length=56695>\n",
      "##contig=<ID=KQ031383.1,length=467143>\n",
      "##contig=<ID=MU273333.1,length=1572686>\n",
      "##contig=<ID=KN538369.1,length=541038>\n",
      "##contig=<ID=MU273382.1,length=187626>\n",
      "##contig=<ID=ML143342.1,length=84043>\n",
      "##contig=<ID=ML143350.1,length=89956>\n",
      "##contig=<ID=ML143362.1,length=192531>\n",
      "##contig=<ID=JH159136.1,length=200998>\n",
      "##contig=<ID=ML143357.1,length=165419>\n",
      "##contig=<ID=ML143385.1,length=17435>\n",
      "##contig=<ID=ML143378.1,length=461303>\n",
      "##contig=<ID=ML143382.1,length=28824>\n",
      "##contig=<ID=ML143383.1,length=68192>\n",
      "##contig=<ID=ML143384.1,length=14678>\n",
      "##contig=<ID=JH159137.1,length=191409>\n",
      "##contig=<ID=ML143356.1,length=45257>\n",
      "##contig=<ID=ML143364.1,length=158944>\n",
      "##contig=<ID=ML143365.1,length=65394>\n",
      "##contig=<ID=MU273369.1,length=434831>\n",
      "##contig=<ID=KZ208923.1,length=48370>\n",
      "##contig=<ID=MU273398.1,length=865743>\n",
      "##contig=<ID=KZ208924.1,length=209722>\n",
      "##contig=<ID=KQ031387.1,length=320750>\n",
      "##contig=<ID=KV766195.1,length=140877>\n",
      "##contig=<ID=KZ208916.1,length=1046838>\n",
      "##contig=<ID=ML143363.1,length=7309>\n",
      "##contig=<ID=KN538360.1,length=460100>\n",
      "##contig=<ID=KZ208920.1,length=690932>\n",
      "##contig=<ID=KZ208906.1,length=330031>\n",
      "##contig=<ID=KN196484.1,length=370917>\n",
      "##contig=<ID=KN196476.1,length=305979>\n",
      "##contig=<ID=KQ983257.1,length=230434>\n",
      "##contig=<ID=KN196479.1,length=330164>\n",
      "##contig=<ID=MU273363.1,length=207371>\n",
      "##contig=<ID=KV575245.1,length=154723>\n",
      "##contig=<ID=KZ208917.1,length=64689>\n",
      "##contig=<ID=MU273343.1,length=489404>\n",
      "##contig=<ID=KZ208911.1,length=242796>\n",
      "##contig=<ID=KN196473.1,length=166200>\n",
      "##contig=<ID=KZ559108.1,length=305244>\n",
      "##contig=<ID=KN196487.1,length=101150>\n",
      "##contig=<ID=KQ759760.1,length=315610>\n",
      "##contig=<ID=KN196475.1,length=451168>\n",
      "##contig=<ID=KV880766.1,length=156998>\n",
      "##contig=<ID=KV880767.1,length=265876>\n",
      "##contig=<ID=MU273348.1,length=475876>\n",
      "##contig=<ID=KQ090016.1,length=245716>\n",
      "##contig=<ID=MU273347.1,length=301310>\n",
      "##contig=<ID=ML143374.1,length=137908>\n",
      "##contig=<ID=KV880764.1,length=142129>\n",
      "##contig=<ID=KN538361.1,length=305542>\n",
      "##contig=<ID=KN196474.1,length=122022>\n",
      "##contig=<ID=ML143360.1,length=170928>\n",
      "##contig=<ID=KZ559109.1,length=279644>\n",
      "##contig=<ID=ML143359.1,length=217075>\n",
      "##contig=<ID=KQ090022.1,length=181958>\n",
      "##contig=<ID=MU273379.1,length=234878>\n",
      "##contig=<ID=KV766194.1,length=139427>\n",
      "##contig=<ID=KN196478.1,length=268330>\n",
      "##contig=<ID=KZ559104.1,length=105527>\n",
      "##contig=<ID=MU273345.1,length=174385>\n",
      "##contig=<ID=MU273350.1,length=113364>\n",
      "##contig=<ID=MU273364.1,length=340717>\n",
      "##contig=<ID=MU273359.1,length=150302>\n",
      "##contig=<ID=KN196480.1,length=277797>\n",
      "##contig=<ID=ML143370.1,length=369264>\n",
      "##contig=<ID=KQ090028.1,length=407387>\n",
      "##contig=<ID=KN196483.1,length=35455>\n",
      "##contig=<ID=KN196481.1,length=108875>\n",
      "##contig=<ID=MU273390.1,length=336752>\n",
      "##contig=<ID=MU273388.1,length=273725>\n",
      "##contig=<ID=MU273344.1,length=244725>\n",
      "##contig=<ID=KN538363.1,length=365499>\n",
      "##contig=<ID=KN538362.1,length=208149>\n",
      "##contig=<ID=KQ031385.1,length=373699>\n",
      "##contig=<ID=KV766192.1,length=411654>\n",
      "##contig=<ID=KQ031386.1,length=165718>\n",
      "##contig=<ID=KQ031388.1,length=179932>\n",
      "##contig=<ID=KN538365.1,length=14347>\n",
      "##contig=<ID=KN538366.1,length=85284>\n",
      "##contig=<ID=KN538367.1,length=420164>\n",
      "##contig=<ID=ML143361.1,length=297568>\n",
      "##contig=<ID=KN538370.1,length=86533>\n",
      "##contig=<ID=KN538373.1,length=148762>\n",
      "##contig=<ID=MU273381.1,length=144689>\n",
      "##contig=<ID=KZ559113.1,length=480415>\n",
      "##contig=<ID=MU273346.1,length=469342>\n",
      "##contig=<ID=MU273391.1,length=1020778>\n",
      "##contig=<ID=KV880765.1,length=468267>\n",
      "##contig=<ID=MU273362.1,length=429744>\n",
      "##contig=<ID=MU273342.1,length=955087>\n",
      "##contig=<ID=MU273374.1,length=1154574>\n",
      "##contig=<ID=KV766196.1,length=281919>\n",
      "##contig=<ID=KN538371.1,length=206320>\n",
      "##contig=<ID=KQ031384.1,length=481245>\n",
      "##contig=<ID=KN538372.1,length=356766>\n",
      "##contig=<ID=MU273355.1,length=508332>\n",
      "##contig=<ID=KQ090021.1,length=264545>\n",
      "##contig=<ID=ML143371.1,length=5500449>\n",
      "##contig=<ID=KN196482.1,length=211377>\n",
      "##contig=<ID=MU273380.1,length=538541>\n",
      "##contig=<ID=MU273361.1,length=106905>\n",
      "##contig=<ID=KZ559115.1,length=230843>\n",
      "##contig=<ID=KZ208914.1,length=165120>\n",
      "##contig=<ID=KZ208922.1,length=93070>\n",
      "##contig=<ID=MU273384.1,length=333754>\n",
      "##contig=<ID=ML143373.1,length=270967>\n",
      "##contig=<ID=MU273353.1,length=208405>\n",
      "##contig=<ID=MU273341.1,length=120381>\n",
      "##contig=<ID=ML143369.1,length=97763>\n",
      "##contig=<ID=ML143366.1,length=409912>\n",
      "##contig=<ID=ML143367.1,length=399183>\n",
      "##contig=<ID=ML143372.1,length=396515>\n",
      "##contig=<ID=ML143380.1,length=412368>\n",
      "##contig=<ID=ML143377.1,length=519485>\n",
      "##contig=<ID=MU273335.1,length=211934>\n",
      "##contig=<ID=MU273392.1,length=189707>\n",
      "##contig=<ID=ML143345.1,length=341066>\n",
      "##contig=<ID=MU273373.1,length=722645>\n",
      "##contig=<ID=MU273393.1,length=68810>\n",
      "##contig=<ID=MU273394.1,length=140567>\n",
      "##contig=<ID=MU273372.1,length=104537>\n",
      "##contig=<ID=MU273370.1,length=344606>\n",
      "##contig=<ID=MU273385.1,length=137818>\n",
      "##contig=<ID=MU273334.1,length=210426>\n",
      "##contig=<ID=MU273367.1,length=196262>\n",
      "##contig=<ID=MU273336.1,length=250447>\n",
      "##contig=<ID=MU273371.1,length=122722>\n",
      "##contig=<ID=MU273383.1,length=172609>\n",
      "##contig=<ID=KQ458386.1,length=405389>\n",
      "##contig=<ID=MU273351.1,length=205691>\n",
      "##contig=<ID=ML143358.1,length=270122>\n",
      "##contig=<ID=KV575244.1,length=673059>\n",
      "##contig=<ID=MU273376.1,length=87715>\n",
      "##contig=<ID=MU273377.1,length=334997>\n",
      "##contig=<ID=MU273389.1,length=355731>\n",
      "##contig=<ID=ML143381.1,length=403128>\n",
      "##contig=<ID=KZ559100.1,length=44955>\n",
      "##contig=<ID=ML143379.1,length=12295>\n",
      "##contig=<ID=ML143354.1,length=454963>\n",
      "##contig=<ID=ML143351.1,length=73265>\n",
      "##contig=<ID=ML143353.1,length=25408>\n",
      "##contig=<ID=ML143344.1,length=235734>\n",
      "##contig=<ID=ML143349.1,length=276109>\n",
      "##contig=<ID=KZ208912.1,length=589656>\n",
      "##contig=<ID=ML143341.1,length=145975>\n",
      "##contig=<ID=KZ208915.1,length=6367528>\n",
      "##contig=<ID=KV880768.1,length=1927115>\n",
      "##contig=<ID=KN196472.1,length=186494>\n",
      "##contig=<ID=GL383545.1,length=179254>\n",
      "##contig=<ID=GL383546.1,length=309802>\n",
      "##contig=<ID=KI270824.1,length=181496>\n",
      "##contig=<ID=KI270825.1,length=188315>\n",
      "##contig=<ID=KQ090020.1,length=185507>\n",
      "##contig=<ID=GL383547.1,length=154407>\n",
      "##contig=<ID=KN538368.1,length=203552>\n",
      "##contig=<ID=KI270826.1,length=186169>\n",
      "##contig=<ID=KI270827.1,length=67707>\n",
      "##contig=<ID=KZ559111.1,length=181167>\n",
      "##contig=<ID=KI270829.1,length=204059>\n",
      "##contig=<ID=KI270830.1,length=177092>\n",
      "##contig=<ID=KI270831.1,length=296895>\n",
      "##contig=<ID=KI270832.1,length=210133>\n",
      "##contig=<ID=KI270902.1,length=106711>\n",
      "##contig=<ID=KI270903.1,length=214625>\n",
      "##contig=<ID=MU273368.1,length=261194>\n",
      "##contig=<ID=KZ559110.1,length=301637>\n",
      "##contig=<ID=KI270927.1,length=218612>\n",
      "##contig=<ID=GL877875.1,length=167313>\n",
      "##contig=<ID=GL383549.1,length=120804>\n",
      "##contig=<ID=GL383550.2,length=169178>\n",
      "##contig=<ID=KQ090023.1,length=109323>\n",
      "##contig=<ID=GL877876.1,length=408271>\n",
      "##contig=<ID=GL383552.1,length=138655>\n",
      "##contig=<ID=KI270904.1,length=572349>\n",
      "##contig=<ID=GL383553.2,length=152874>\n",
      "##contig=<ID=KI270835.1,length=238139>\n",
      "##contig=<ID=GL383551.1,length=184319>\n",
      "##contig=<ID=KI270837.1,length=40090>\n",
      "##contig=<ID=KI270833.1,length=76061>\n",
      "##contig=<ID=KI270834.1,length=119498>\n",
      "##contig=<ID=KI270836.1,length=56134>\n",
      "##contig=<ID=KZ208918.1,length=174808>\n",
      "##contig=<ID=KZ559112.1,length=154139>\n",
      "##contig=<ID=KI270838.1,length=306913>\n",
      "##contig=<ID=KI270839.1,length=180306>\n",
      "##contig=<ID=KI270840.1,length=191684>\n",
      "##contig=<ID=KI270841.1,length=169134>\n",
      "##contig=<ID=KI270842.1,length=37287>\n",
      "##contig=<ID=KI270843.1,length=103832>\n",
      "##contig=<ID=KQ090024.1,length=168146>\n",
      "##contig=<ID=KQ090025.1,length=123480>\n",
      "##contig=<ID=KI270844.1,length=322166>\n",
      "##contig=<ID=KI270845.1,length=180703>\n",
      "##contig=<ID=KI270846.1,length=1351393>\n",
      "##contig=<ID=KI270847.1,length=1511111>\n",
      "##contig=<ID=KZ208919.1,length=171798>\n",
      "##contig=<ID=ML143368.1,length=264228>\n",
      "##contig=<ID=KI270852.1,length=478999>\n",
      "##contig=<ID=KI270848.1,length=327382>\n",
      "##contig=<ID=GL383554.1,length=296527>\n",
      "##contig=<ID=KI270906.1,length=196384>\n",
      "##contig=<ID=GL383555.2,length=388773>\n",
      "##contig=<ID=KI270851.1,length=263054>\n",
      "##contig=<ID=KI270849.1,length=244917>\n",
      "##contig=<ID=KI270905.1,length=5161414>\n",
      "##contig=<ID=KI270850.1,length=430880>\n",
      "##contig=<ID=KQ031389.1,length=2365364>\n",
      "##contig=<ID=MU273375.1,length=204007>\n",
      "##contig=<ID=KI270853.1,length=2659700>\n",
      "##contig=<ID=GL383556.1,length=192462>\n",
      "##contig=<ID=GL383557.1,length=89672>\n",
      "##contig=<ID=KI270855.1,length=232857>\n",
      "##contig=<ID=KQ031390.1,length=169136>\n",
      "##contig=<ID=KI270856.1,length=63982>\n",
      "##contig=<ID=KQ090027.1,length=267463>\n",
      "##contig=<ID=KQ090026.1,length=59016>\n",
      "##contig=<ID=KZ208921.1,length=78609>\n",
      "##contig=<ID=KI270854.1,length=134193>\n",
      "##contig=<ID=KI270909.1,length=325800>\n",
      "##contig=<ID=KV766197.1,length=246895>\n",
      "##contig=<ID=KZ559114.1,length=116753>\n",
      "##contig=<ID=MU273378.1,length=372839>\n",
      "##contig=<ID=GL383563.3,length=375691>\n",
      "##contig=<ID=KI270861.1,length=196688>\n",
      "##contig=<ID=GL383564.2,length=133151>\n",
      "##contig=<ID=GL000258.2,length=1821992>\n",
      "##contig=<ID=KI270860.1,length=178921>\n",
      "##contig=<ID=KI270907.1,length=137721>\n",
      "##contig=<ID=KI270862.1,length=391357>\n",
      "##contig=<ID=GL383565.1,length=223995>\n",
      "##contig=<ID=KI270908.1,length=1423190>\n",
      "##contig=<ID=KV766198.1,length=276292>\n",
      "##contig=<ID=KI270910.1,length=157099>\n",
      "##contig=<ID=GL383566.1,length=90219>\n",
      "##contig=<ID=JH159146.1,length=278131>\n",
      "##contig=<ID=JH159147.1,length=70345>\n",
      "##contig=<ID=JH159148.1,length=88070>\n",
      "##contig=<ID=KI270857.1,length=2877074>\n",
      "##contig=<ID=KI270858.1,length=235827>\n",
      "##contig=<ID=KI270859.1,length=108763>\n",
      "##contig=<ID=KZ559116.1,length=163186>\n",
      "##contig=<ID=GL383567.1,length=289831>\n",
      "##contig=<ID=GL383568.1,length=104552>\n",
      "##contig=<ID=GL383569.1,length=167950>\n",
      "##contig=<ID=GL383570.1,length=164789>\n",
      "##contig=<ID=GL383571.1,length=198278>\n",
      "##contig=<ID=GL383572.1,length=159547>\n",
      "##contig=<ID=KI270863.1,length=167999>\n",
      "##contig=<ID=KI270864.1,length=111737>\n",
      "##contig=<ID=KQ458385.1,length=205101>\n",
      "##contig=<ID=KI270912.1,length=174061>\n",
      "##contig=<ID=KI270911.1,length=157710>\n",
      "##contig=<ID=KV575254.1,length=99845>\n",
      "##contig=<ID=KV575246.1,length=163926>\n",
      "##contig=<ID=KV575256.1,length=223118>\n",
      "##contig=<ID=KV575253.1,length=166713>\n",
      "##contig=<ID=KV575252.1,length=178197>\n",
      "##contig=<ID=KV575255.1,length=161095>\n",
      "##contig=<ID=KV575259.1,length=171263>\n",
      "##contig=<ID=KI270917.1,length=190932>\n",
      "##contig=<ID=KI270918.1,length=123111>\n",
      "##contig=<ID=KI270919.1,length=170701>\n",
      "##contig=<ID=KV575247.1,length=170206>\n",
      "##contig=<ID=KV575248.1,length=168131>\n",
      "##contig=<ID=KV575250.1,length=241058>\n",
      "##contig=<ID=KV575249.1,length=293522>\n",
      "##contig=<ID=KV575257.1,length=100553>\n",
      "##contig=<ID=KI270920.1,length=198005>\n",
      "##contig=<ID=KI270921.1,length=282224>\n",
      "##contig=<ID=KI270922.1,length=187935>\n",
      "##contig=<ID=KI270923.1,length=189352>\n",
      "##contig=<ID=KI270929.1,length=186203>\n",
      "##contig=<ID=KI270930.1,length=200773>\n",
      "##contig=<ID=KI270931.1,length=170148>\n",
      "##contig=<ID=KI270932.1,length=215732>\n",
      "##contig=<ID=KI270933.1,length=170537>\n",
      "##contig=<ID=KI270882.1,length=248807>\n",
      "##contig=<ID=KI270883.1,length=170399>\n",
      "##contig=<ID=KI270884.1,length=157053>\n",
      "##contig=<ID=KI270885.1,length=171027>\n",
      "##contig=<ID=KI270886.1,length=204239>\n",
      "##contig=<ID=KI270887.1,length=209512>\n",
      "##contig=<ID=KI270888.1,length=155532>\n",
      "##contig=<ID=KV575258.1,length=156965>\n",
      "##contig=<ID=KV575251.1,length=159285>\n",
      "##contig=<ID=KV575260.1,length=145691>\n",
      "##contig=<ID=KI270889.1,length=170698>\n",
      "##contig=<ID=KI270890.1,length=184499>\n",
      "##contig=<ID=GL000209.2,length=177381>\n",
      "##contig=<ID=KI270891.1,length=170680>\n",
      "##contig=<ID=KI270914.1,length=205194>\n",
      "##contig=<ID=KI270915.1,length=170665>\n",
      "##contig=<ID=KI270916.1,length=184516>\n",
      "##contig=<ID=GL949746.1,length=987716>\n",
      "##contig=<ID=GL949747.2,length=729520>\n",
      "##contig=<ID=GL949748.2,length=1064304>\n",
      "##contig=<ID=GL949749.2,length=1091841>\n",
      "##contig=<ID=GL949750.2,length=1066390>\n",
      "##contig=<ID=GL949751.2,length=1002683>\n",
      "##contig=<ID=GL949752.1,length=987100>\n",
      "##contig=<ID=GL949753.2,length=796479>\n",
      "##contig=<ID=GL383573.1,length=385657>\n",
      "##contig=<ID=GL383574.1,length=155864>\n",
      "##contig=<ID=GL383575.2,length=170222>\n",
      "##contig=<ID=KI270866.1,length=43156>\n",
      "##contig=<ID=GL383576.1,length=188024>\n",
      "##contig=<ID=KI270867.1,length=233762>\n",
      "##contig=<ID=KI270865.1,length=52969>\n",
      "##contig=<ID=KI270938.1,length=1066800>\n",
      "##contig=<ID=KI270868.1,length=61734>\n",
      "##contig=<ID=MU273387.1,length=89211>\n",
      "##contig=<ID=MU273330.1,length=516764>\n",
      "##contig=<ID=KI270760.1,length=109528>\n",
      "##contig=<ID=KI270762.1,length=354444>\n",
      "##contig=<ID=GL383518.1,length=182439>\n",
      "##contig=<ID=KI270759.1,length=425601>\n",
      "##contig=<ID=KI270766.1,length=256271>\n",
      "##contig=<ID=GL383519.1,length=110268>\n",
      "##contig=<ID=KI270761.1,length=165834>\n",
      "##contig=<ID=KQ458382.1,length=141019>\n",
      "##contig=<ID=GL383520.2,length=366580>\n",
      "##contig=<ID=KI270763.1,length=911658>\n",
      "##contig=<ID=KQ458383.1,length=349938>\n",
      "##contig=<ID=KI270765.1,length=185285>\n",
      "##contig=<ID=KI270764.1,length=50258>\n",
      "##contig=<ID=KQ983255.1,length=278659>\n",
      "##contig=<ID=MU273331.1,length=847441>\n",
      "##contig=<ID=KQ458384.1,length=212205>\n",
      "##contig=<ID=KV880763.1,length=551020>\n",
      "##contig=<ID=MU273332.1,length=335159>\n",
      "##contig=<ID=KZ208904.1,length=166136>\n",
      "##contig=<ID=KZ208905.1,length=140355>\n",
      "##contig=<ID=KI270892.1,length=162212>\n",
      "##contig=<ID=GL383577.2,length=128386>\n",
      "##contig=<ID=KI270869.1,length=118774>\n",
      "##contig=<ID=KI270870.1,length=183433>\n",
      "##contig=<ID=KI270871.1,length=58661>\n",
      "##contig=<ID=GL383578.2,length=63917>\n",
      "##contig=<ID=GL383579.2,length=201197>\n",
      "##contig=<ID=GL383580.2,length=74653>\n",
      "##contig=<ID=GL383581.2,length=116689>\n",
      "##contig=<ID=KI270872.1,length=82692>\n",
      "##contig=<ID=KI270873.1,length=143900>\n",
      "##contig=<ID=KI270874.1,length=166743>\n",
      "##contig=<ID=GL383582.2,length=162811>\n",
      "##contig=<ID=GL383583.2,length=96924>\n",
      "##contig=<ID=KI270875.1,length=259914>\n",
      "##contig=<ID=KI270876.1,length=263666>\n",
      "##contig=<ID=KI270877.1,length=101331>\n",
      "##contig=<ID=KI270878.1,length=186262>\n",
      "##contig=<ID=KI270879.1,length=304135>\n",
      "##contig=<ID=KB663609.1,length=74013>\n",
      "##contig=<ID=KI270928.1,length=176103>\n",
      "##contig=<ID=KN196485.1,length=156562>\n",
      "##contig=<ID=KN196486.1,length=153027>\n",
      "##contig=<ID=KQ458387.1,length=155930>\n",
      "##contig=<ID=KQ458388.1,length=174749>\n",
      "##contig=<ID=KQ759761.1,length=145162>\n",
      "##contig=<ID=MU273337.1,length=431782>\n",
      "##contig=<ID=MU273338.1,length=535251>\n",
      "##contig=<ID=MU273339.1,length=500581>\n",
      "##contig=<ID=KI270769.1,length=120616>\n",
      "##contig=<ID=KI270767.1,length=161578>\n",
      "##contig=<ID=GL383521.1,length=143390>\n",
      "##contig=<ID=KI270772.1,length=133041>\n",
      "##contig=<ID=GL383522.1,length=123821>\n",
      "##contig=<ID=KI270770.1,length=136240>\n",
      "##contig=<ID=KI270893.1,length=161218>\n",
      "##contig=<ID=KI270894.1,length=214158>\n",
      "##contig=<ID=GL582966.2,length=96131>\n",
      "##contig=<ID=KI270773.1,length=70887>\n",
      "##contig=<ID=KI270776.1,length=174166>\n",
      "##contig=<ID=KI270768.1,length=110099>\n",
      "##contig=<ID=KI270774.1,length=223625>\n",
      "##contig=<ID=KI270771.1,length=110395>\n",
      "##contig=<ID=KI270775.1,length=138019>\n",
      "##contig=<ID=MU273340.1,length=284971>\n",
      "##contig=<ID=KQ983256.1,length=535088>\n",
      "##contig=<ID=KZ208907.1,length=181658>\n",
      "##contig=<ID=KZ208908.1,length=140361>\n",
      "##contig=<ID=JH636055.2,length=173151>\n",
      "##contig=<ID=GL383526.1,length=180671>\n",
      "##contig=<ID=KI270779.1,length=205312>\n",
      "##contig=<ID=KI270777.1,length=173649>\n",
      "##contig=<ID=KI270782.1,length=162429>\n",
      "##contig=<ID=KI270783.1,length=109187>\n",
      "##contig=<ID=KI270778.1,length=248252>\n",
      "##contig=<ID=KI270895.1,length=162896>\n",
      "##contig=<ID=KZ208909.1,length=175849>\n",
      "##contig=<ID=KI270780.1,length=224108>\n",
      "##contig=<ID=KI270924.1,length=166540>\n",
      "##contig=<ID=ML143343.1,length=215443>\n",
      "##contig=<ID=KI270781.1,length=113034>\n",
      "##contig=<ID=KI270934.1,length=163458>\n",
      "##contig=<ID=KZ559105.1,length=195063>\n",
      "##contig=<ID=KI270935.1,length=197351>\n",
      "##contig=<ID=KZ559101.1,length=164041>\n",
      "##contig=<ID=KI270936.1,length=164170>\n",
      "##contig=<ID=KZ559102.1,length=197752>\n",
      "##contig=<ID=KI270937.1,length=165607>\n",
      "##contig=<ID=KZ559103.1,length=302885>\n",
      "##contig=<ID=KI270784.1,length=184404>\n",
      "##contig=<ID=KQ983258.1,length=205407>\n",
      "##contig=<ID=KV766193.1,length=420675>\n",
      "##contig=<ID=GL383527.1,length=164536>\n",
      "##contig=<ID=KI270790.1,length=220246>\n",
      "##contig=<ID=GL383528.1,length=376187>\n",
      "##contig=<ID=KI270787.1,length=111943>\n",
      "##contig=<ID=GL000257.2,length=586476>\n",
      "##contig=<ID=KI270785.1,length=119912>\n",
      "##contig=<ID=KQ090013.1,length=90922>\n",
      "##contig=<ID=MU273349.1,length=308682>\n",
      "##contig=<ID=KI270786.1,length=244096>\n",
      "##contig=<ID=KI270788.1,length=158965>\n",
      "##contig=<ID=KI270789.1,length=205944>\n",
      "##contig=<ID=KI270896.1,length=378547>\n",
      "##contig=<ID=KI270925.1,length=555799>\n",
      "##contig=<ID=KQ090014.1,length=163749>\n",
      "##contig=<ID=KQ090015.1,length=236512>\n",
      "##contig=<ID=MU273356.1,length=302485>\n",
      "##contig=<ID=GL383532.1,length=82728>\n",
      "##contig=<ID=KI270897.1,length=1144418>\n",
      "##contig=<ID=GL383531.1,length=173459>\n",
      "##contig=<ID=GL949742.1,length=226852>\n",
      "##contig=<ID=GL339449.2,length=1612928>\n",
      "##contig=<ID=KI270795.1,length=131892>\n",
      "##contig=<ID=KI270791.1,length=195710>\n",
      "##contig=<ID=GL383530.1,length=101241>\n",
      "##contig=<ID=KI270898.1,length=130957>\n",
      "##contig=<ID=KI270792.1,length=179043>\n",
      "##contig=<ID=KI270796.1,length=172708>\n",
      "##contig=<ID=KI270793.1,length=126136>\n",
      "##contig=<ID=KI270794.1,length=164558>\n",
      "##contig=<ID=KN196477.1,length=139087>\n",
      "##contig=<ID=KV575243.1,length=362221>\n",
      "##contig=<ID=KZ208910.1,length=135987>\n",
      "##contig=<ID=MU273357.1,length=383128>\n",
      "##contig=<ID=KQ090017.1,length=82315>\n",
      "##contig=<ID=GL383533.1,length=124736>\n",
      "##contig=<ID=KB021644.2,length=185823>\n",
      "##contig=<ID=KI270797.1,length=197536>\n",
      "##contig=<ID=KI270798.1,length=271782>\n",
      "##contig=<ID=KI270799.1,length=152148>\n",
      "##contig=<ID=KI270800.1,length=175808>\n",
      "##contig=<ID=KI270801.1,length=870480>\n",
      "##contig=<ID=KI270802.1,length=75005>\n",
      "##contig=<ID=KI270758.1,length=76752>\n",
      "##contig=<ID=GL000250.2,length=4672374>\n",
      "##contig=<ID=GL000251.2,length=4795265>\n",
      "##contig=<ID=GL000252.2,length=4604811>\n",
      "##contig=<ID=GL000253.2,length=4677643>\n",
      "##contig=<ID=GL000254.2,length=4827813>\n",
      "##contig=<ID=GL000255.2,length=4606388>\n",
      "##contig=<ID=GL000256.2,length=4929269>\n",
      "##contig=<ID=KI270804.1,length=157952>\n",
      "##contig=<ID=KI270806.1,length=158166>\n",
      "##contig=<ID=GL383534.2,length=119183>\n",
      "##contig=<ID=KI270805.1,length=209988>\n",
      "##contig=<ID=KI270899.1,length=190869>\n",
      "##contig=<ID=KI270809.1,length=209586>\n",
      "##contig=<ID=KI270803.1,length=1111570>\n",
      "##contig=<ID=KI270807.1,length=126434>\n",
      "##contig=<ID=KZ559106.1,length=172555>\n",
      "##contig=<ID=KZ208913.1,length=680662>\n",
      "##contig=<ID=KI270808.1,length=271455>\n",
      "##contig=<ID=MU273358.1,length=464417>\n",
      "##contig=<ID=KI270811.1,length=292436>\n",
      "##contig=<ID=KI270814.1,length=141812>\n",
      "##contig=<ID=KI270810.1,length=374415>\n",
      "##contig=<ID=KI270812.1,length=282736>\n",
      "##contig=<ID=KI270815.1,length=132244>\n",
      "##contig=<ID=KI270813.1,length=300230>\n",
      "##contig=<ID=KI270816.1,length=305841>\n",
      "##contig=<ID=KI270818.1,length=145606>\n",
      "##contig=<ID=KI270817.1,length=158983>\n",
      "##contig=<ID=KI270900.1,length=318687>\n",
      "##contig=<ID=KI270819.1,length=133535>\n",
      "##contig=<ID=KI270901.1,length=136959>\n",
      "##contig=<ID=KI270820.1,length=36640>\n",
      "##contig=<ID=KI270926.1,length=229282>\n",
      "##contig=<ID=KZ559107.1,length=103072>\n",
      "##contig=<ID=KI270821.1,length=985506>\n",
      "##contig=<ID=KI270822.1,length=624492>\n",
      "##contig=<ID=GL383539.1,length=162988>\n",
      "##contig=<ID=GL383540.1,length=71551>\n",
      "##contig=<ID=GL383541.1,length=171286>\n",
      "##contig=<ID=GL383542.1,length=60032>\n",
      "##contig=<ID=KI270823.1,length=439082>\n",
      "##contig=<ID=KQ090018.1,length=163882>\n",
      "##contig=<ID=KQ090019.1,length=134099>\n",
      "##contig=<ID=MU273395.1,length=619716>\n",
      "##contig=<ID=KI270880.1,length=284869>\n",
      "##contig=<ID=KI270881.1,length=144206>\n",
      "##contig=<ID=MU273396.1,length=294119>\n",
      "##contig=<ID=KI270913.1,length=274009>\n",
      "##contig=<ID=MU273397.1,length=330493>\n",
      "##contig=<ID=KV766199.1,length=188004>\n",
      "##contig=<ID=GL000008.2,length=209709>\n",
      "##contig=<ID=GL000009.2,length=201709>\n",
      "##contig=<ID=GL000194.1,length=191469>\n",
      "##contig=<ID=GL000195.1,length=182896>\n",
      "##contig=<ID=GL000205.2,length=185591>\n",
      "##contig=<ID=GL000208.1,length=92689>\n",
      "##contig=<ID=GL000213.1,length=164239>\n",
      "##contig=<ID=GL000214.1,length=137718>\n",
      "##contig=<ID=GL000216.2,length=176608>\n",
      "##contig=<ID=GL000218.1,length=161147>\n",
      "##contig=<ID=GL000219.1,length=179198>\n",
      "##contig=<ID=GL000220.1,length=161802>\n",
      "##contig=<ID=GL000221.1,length=155397>\n",
      "##contig=<ID=GL000224.1,length=179693>\n",
      "##contig=<ID=GL000225.1,length=211173>\n",
      "##contig=<ID=GL000226.1,length=15008>\n",
      "##contig=<ID=KI270302.1,length=2274>\n",
      "##contig=<ID=KI270303.1,length=1942>\n",
      "##contig=<ID=KI270304.1,length=2165>\n",
      "##contig=<ID=KI270305.1,length=1472>\n",
      "##contig=<ID=KI270310.1,length=1201>\n",
      "##contig=<ID=KI270311.1,length=12399>\n",
      "##contig=<ID=KI270312.1,length=998>\n",
      "##contig=<ID=KI270315.1,length=2276>\n",
      "##contig=<ID=KI270316.1,length=1444>\n",
      "##contig=<ID=KI270317.1,length=37690>\n",
      "##contig=<ID=KI270320.1,length=4416>\n",
      "##contig=<ID=KI270322.1,length=21476>\n",
      "##contig=<ID=KI270329.1,length=1040>\n",
      "##contig=<ID=KI270330.1,length=1652>\n",
      "##contig=<ID=KI270333.1,length=2699>\n",
      "##contig=<ID=KI270334.1,length=1368>\n",
      "##contig=<ID=KI270335.1,length=1048>\n",
      "##contig=<ID=KI270336.1,length=1026>\n",
      "##contig=<ID=KI270337.1,length=1121>\n",
      "##contig=<ID=KI270338.1,length=1428>\n",
      "##contig=<ID=KI270340.1,length=1428>\n",
      "##contig=<ID=KI270362.1,length=3530>\n",
      "##contig=<ID=KI270363.1,length=1803>\n",
      "##contig=<ID=KI270364.1,length=2855>\n",
      "##contig=<ID=KI270366.1,length=8320>\n",
      "##contig=<ID=KI270371.1,length=2805>\n",
      "##contig=<ID=KI270372.1,length=1650>\n",
      "##contig=<ID=KI270373.1,length=1451>\n",
      "##contig=<ID=KI270374.1,length=2656>\n",
      "##contig=<ID=KI270375.1,length=2378>\n",
      "##contig=<ID=KI270376.1,length=1136>\n",
      "##contig=<ID=KI270378.1,length=1048>\n",
      "##contig=<ID=KI270379.1,length=1045>\n",
      "##contig=<ID=KI270381.1,length=1930>\n",
      "##contig=<ID=KI270382.1,length=4215>\n",
      "##contig=<ID=KI270383.1,length=1750>\n",
      "##contig=<ID=KI270384.1,length=1658>\n",
      "##contig=<ID=KI270385.1,length=990>\n",
      "##contig=<ID=KI270386.1,length=1788>\n",
      "##contig=<ID=KI270387.1,length=1537>\n",
      "##contig=<ID=KI270388.1,length=1216>\n",
      "##contig=<ID=KI270389.1,length=1298>\n",
      "##contig=<ID=KI270390.1,length=2387>\n",
      "##contig=<ID=KI270391.1,length=1484>\n",
      "##contig=<ID=KI270392.1,length=971>\n",
      "##contig=<ID=KI270393.1,length=1308>\n",
      "##contig=<ID=KI270394.1,length=970>\n",
      "##contig=<ID=KI270395.1,length=1143>\n",
      "##contig=<ID=KI270396.1,length=1880>\n",
      "##contig=<ID=KI270411.1,length=2646>\n",
      "##contig=<ID=KI270412.1,length=1179>\n",
      "##contig=<ID=KI270414.1,length=2489>\n",
      "##contig=<ID=KI270417.1,length=2043>\n",
      "##contig=<ID=KI270418.1,length=2145>\n",
      "##contig=<ID=KI270419.1,length=1029>\n",
      "##contig=<ID=KI270420.1,length=2321>\n",
      "##contig=<ID=KI270422.1,length=1445>\n",
      "##contig=<ID=KI270423.1,length=981>\n",
      "##contig=<ID=KI270424.1,length=2140>\n",
      "##contig=<ID=KI270425.1,length=1884>\n",
      "##contig=<ID=KI270429.1,length=1361>\n",
      "##contig=<ID=KI270435.1,length=92983>\n",
      "##contig=<ID=KI270438.1,length=112505>\n",
      "##contig=<ID=KI270442.1,length=392061>\n",
      "##contig=<ID=KI270448.1,length=7992>\n",
      "##contig=<ID=KI270465.1,length=1774>\n",
      "##contig=<ID=KI270466.1,length=1233>\n",
      "##contig=<ID=KI270467.1,length=3920>\n",
      "##contig=<ID=KI270468.1,length=4055>\n",
      "##contig=<ID=KI270507.1,length=5353>\n",
      "##contig=<ID=KI270508.1,length=1951>\n",
      "##contig=<ID=KI270509.1,length=2318>\n",
      "##contig=<ID=KI270510.1,length=2415>\n",
      "##contig=<ID=KI270511.1,length=8127>\n",
      "##contig=<ID=KI270512.1,length=22689>\n",
      "##contig=<ID=KI270515.1,length=6361>\n",
      "##contig=<ID=KI270516.1,length=1300>\n",
      "##contig=<ID=KI270517.1,length=3253>\n",
      "##contig=<ID=KI270518.1,length=2186>\n",
      "##contig=<ID=KI270519.1,length=138126>\n",
      "##contig=<ID=KI270521.1,length=7642>\n",
      "##contig=<ID=KI270522.1,length=5674>\n",
      "##contig=<ID=KI270528.1,length=2983>\n",
      "##contig=<ID=KI270529.1,length=1899>\n",
      "##contig=<ID=KI270530.1,length=2168>\n",
      "##contig=<ID=KI270538.1,length=91309>\n",
      "##contig=<ID=KI270539.1,length=993>\n",
      "##contig=<ID=KI270544.1,length=1202>\n",
      "##contig=<ID=KI270548.1,length=1599>\n",
      "##contig=<ID=KI270579.1,length=31033>\n",
      "##contig=<ID=KI270580.1,length=1553>\n",
      "##contig=<ID=KI270581.1,length=7046>\n",
      "##contig=<ID=KI270582.1,length=6504>\n",
      "##contig=<ID=KI270583.1,length=1400>\n",
      "##contig=<ID=KI270584.1,length=4513>\n",
      "##contig=<ID=KI270587.1,length=2969>\n",
      "##contig=<ID=KI270588.1,length=6158>\n",
      "##contig=<ID=KI270589.1,length=44474>\n",
      "##contig=<ID=KI270590.1,length=4685>\n",
      "##contig=<ID=KI270591.1,length=5796>\n",
      "##contig=<ID=KI270593.1,length=3041>\n",
      "##contig=<ID=KI270706.1,length=175055>\n",
      "##contig=<ID=KI270707.1,length=32032>\n",
      "##contig=<ID=KI270708.1,length=127682>\n",
      "##contig=<ID=KI270709.1,length=66860>\n",
      "##contig=<ID=KI270710.1,length=40176>\n",
      "##contig=<ID=KI270711.1,length=42210>\n",
      "##contig=<ID=KI270712.1,length=176043>\n",
      "##contig=<ID=KI270713.1,length=40745>\n",
      "##contig=<ID=KI270714.1,length=41717>\n",
      "##contig=<ID=KI270715.1,length=161471>\n",
      "##contig=<ID=KI270716.1,length=153799>\n",
      "##contig=<ID=KI270717.1,length=40062>\n",
      "##contig=<ID=KI270718.1,length=38054>\n",
      "##contig=<ID=KI270719.1,length=176845>\n",
      "##contig=<ID=KI270720.1,length=39050>\n",
      "##contig=<ID=KI270721.1,length=100316>\n",
      "##contig=<ID=KI270722.1,length=194050>\n",
      "##contig=<ID=KI270723.1,length=38115>\n",
      "##contig=<ID=KI270724.1,length=39555>\n",
      "##contig=<ID=KI270725.1,length=172810>\n",
      "##contig=<ID=KI270726.1,length=43739>\n",
      "##contig=<ID=KI270727.1,length=448248>\n",
      "##contig=<ID=KI270728.1,length=1872759>\n",
      "##contig=<ID=KI270729.1,length=280839>\n",
      "##contig=<ID=KI270730.1,length=112551>\n",
      "##contig=<ID=KI270731.1,length=150754>\n",
      "##contig=<ID=KI270732.1,length=41543>\n",
      "##contig=<ID=KI270733.1,length=179772>\n",
      "##contig=<ID=KI270734.1,length=165050>\n",
      "##contig=<ID=KI270735.1,length=42811>\n",
      "##contig=<ID=KI270736.1,length=181920>\n",
      "##contig=<ID=KI270737.1,length=103838>\n",
      "##contig=<ID=KI270738.1,length=99375>\n",
      "##contig=<ID=KI270739.1,length=73985>\n",
      "##contig=<ID=KI270740.1,length=37240>\n",
      "##contig=<ID=KI270741.1,length=157432>\n",
      "##contig=<ID=KI270742.1,length=186739>\n",
      "##contig=<ID=KI270743.1,length=210658>\n",
      "##contig=<ID=KI270744.1,length=168472>\n",
      "##contig=<ID=KI270745.1,length=41891>\n",
      "##contig=<ID=KI270746.1,length=66486>\n",
      "##contig=<ID=KI270747.1,length=198735>\n",
      "##contig=<ID=KI270748.1,length=93321>\n",
      "##contig=<ID=KI270749.1,length=158759>\n",
      "##contig=<ID=KI270750.1,length=148850>\n",
      "##contig=<ID=KI270751.1,length=150742>\n",
      "##contig=<ID=KI270752.1,length=27745>\n",
      "##contig=<ID=KI270753.1,length=62944>\n",
      "##contig=<ID=KI270754.1,length=40191>\n",
      "##contig=<ID=KI270755.1,length=36723>\n",
      "##contig=<ID=KI270756.1,length=79590>\n",
      "##contig=<ID=KI270757.1,length=71251>\n",
      "##source=HaplotypeCaller\n",
      "##bcftools_viewVersion=1.22+htslib-1.22.1\n",
      "##bcftools_viewCommand=view -R /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/egfr_exons.3col.bed /home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.egfr.del19.vcf.gz; Date=Sat Jul 19 11:53:11 2025\n",
      "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tPAR-UT\n"
     ]
    }
   ],
   "source": [
    "! bcftools view -R \"/home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/egfr_exons.3col.bed\" \"/home/creixell/AXLomics/msresist/data/RNAseq/EGFRmutations/Res_PC9-PAR-UT/PAR-UT.egfr.del19.vcf.gz\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pysradb",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
